<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Intro &mdash; AgBase 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="InterProScan on CyVerse" href="using_iprs_in_cyverse_DE.html" />
    <link rel="prev" title="GOanna on the Command Line" href="../goanna/using_goanna_cmd.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            AgBase
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Agbase Home</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../agbase/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">InterProScan</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Intro</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#where-to-find-interproscan"><strong>Where to Find InterProScan</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-interproscan-data"><strong>Getting the InterProScan Data</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#help-and-usage-statement"><strong>Help and Usage Statement</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="using_iprs_in_cyverse_DE.html"><strong>InterProScan on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Combine GAFs</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cyverse.html"><strong>Combine GAFs on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cmd.html"><strong>Combine GAFs on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">KOBAS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kobas/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_cmd.html"><strong>KOBAS on the Command Line</strong></a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">AgBase</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active"><strong>Intro</strong></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/interproscan/intro.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="intro">
<h1><strong>Intro</strong><a class="headerlink" href="#intro" title="Permalink to this headline"></a></h1>
<p><a class="reference external" href="http://www.ebi.ac.uk/interpro/">InterPro</a> is a database which integrates together predictive information about proteins’ function from a number of partner resources, giving an overview of the families that a protein belongs to and the domains and sites it contains.</p>
<p><strong>Basic functions of this tool</strong></p>
<ul class="simple">
<li><p>removes special characters from FASTA sequences</p></li>
<li><p>splits FASTA into groups of 1000 sequences</p></li>
<li><p>runs InterProScan with user-specified options on each of the 1000-sequence files in parallel</p></li>
<li><p>re-combines output files from all groups of 1000</p></li>
<li><p>parses the XML output from InterProScan to generate a gene association file (GAF) (and several other files)</p></li>
</ul>
<p>Results and analysis from the application of InterProScan annotation to the Official gene set v3.0 protein set from <em>Diaphorina citri</em> followed by a differential expression analysis was presented at a seminar in the University of Arizona Animal and Comparative Biomedical Sciences in Fall 2020. The <a class="reference external" href="https://www.slideshare.net/suryasaha/functional-annotation-of-invertebrate-genomes">slides</a> and <a class="reference external" href="https://arizona.zoom.us/rec/play/tZZ-fuutrj43T9fBtASDAaR9W9S0fP6s1XQbrvQOz0e0VnYHYVL1MOMaZ-F4v45qOmXQkV1MUXQ7tufD">video</a> are available online.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tool accepts a peptide FASTA file. For those users with nucloetide sequences some documentation has been provided for using <strong>TransDecoder</strong> (although other tools are also acceptable).
The <a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=74828a18-f351-11e8-be2b-008cfa5ae621&amp;system-id=de">TransDecoder app</a> is available through CyVerse or as a <a class="reference external" href="https://quay.io/repository/biocontainers/transdecoder?tab=tags">BioContainer</a> for use on the command line.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As both GOanna and InterProScan provide GO annotations, their outputs are provided in GAF format. The <strong>‘Combine GAFs’</strong> tool can then be used to make a single GAF of GO annotations, if desired.</p>
</div>
<section id="where-to-find-interproscan">
<h2><strong>Where to Find InterProScan</strong><a class="headerlink" href="#where-to-find-interproscan" title="Permalink to this headline"></a></h2>
<p><a class="reference external" href="https://hub.docker.com/r/agbase/interproscan">Docker Hub (5.63-95)</a></p>
<p><a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=Interproscan-5.36.75u2&amp;system-id=agave">CyVerse (5.36-75)</a></p>
</section>
<section id="getting-the-interproscan-data">
<h2><strong>Getting the InterProScan Data</strong><a class="headerlink" href="#getting-the-interproscan-data" title="Permalink to this headline"></a></h2>
<p><strong>InterProScan Data (now includes Panther)</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.63-95.0/alt/interproscan-data-5.63-95.0.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.63-95.0/alt/interproscan-data-5.63-95.0.tar.gz.md5
md5sum -c interproscan-data-5.63-95.0.tar.gz.md
tar -pxvzf interproscan-data-5.63-795.0.tar.gz
</pre></div>
</div>
<div class="admonition-tar-options admonition">
<p class="admonition-title">tar options</p>
<ul class="simple">
<li><p>p = preserve the file permissions</p></li>
<li><p>x = extract files from an archive</p></li>
<li><p>v = verbosely list the files processed</p></li>
<li><p>z = filter the archive through gzip</p></li>
<li><p>f = use archive file</p></li>
</ul>
</div>
</section>
<section id="help-and-usage-statement">
<span id="iprsusage"></span><h2><strong>Help and Usage Statement</strong><a class="headerlink" href="#help-and-usage-statement" title="Permalink to this headline"></a></h2>
<div class="highlight-bash notranslate" id="id1"><div class="highlight"><pre><span></span>   Options:
-a  &lt;ANALYSES&gt;                             Optional, comma separated list of analyses.  If this option
                                           is not set, ALL analyses will be run.

-b &lt;OUTPUT-FILE-BASE&gt;                     Optional, base output filename <span class="o">(</span>relative or absolute path<span class="o">)</span>.
                                           Note that this option, the output directory <span class="o">(</span>-d<span class="o">)</span> option and
                                           the output file name <span class="o">(</span>-o<span class="o">)</span> option are mutually exclusive.  The
                                           appropriate file extension <span class="k">for</span> the output format<span class="o">(</span>s<span class="o">)</span> will be
                                           appended automatically. By default the input file
                                           path/name will be used.

-d &lt;OUTPUT-DIR&gt;                            Optional, output directory. Note that this option, the
                                           output file name <span class="o">(</span>-o<span class="o">)</span> option and the output file base <span class="o">(</span>-b<span class="o">)</span> option
                                           are mutually exclusive. The output filename<span class="o">(</span>s<span class="o">)</span> are the
                                           same as the input filename, with the appropriate file
                                           extension<span class="o">(</span>s<span class="o">)</span> <span class="k">for</span> the output format<span class="o">(</span>s<span class="o">)</span> appended automatically .

-c                                         Optional.  Disables use of the precalculated match lookup
                                           service.  All match calculations will be run locally.

-C                                         Optional. Supply the number of cpus to use.

-e                                         Optional, excludes sites from the XML, JSON output

-f &lt;OUTPUT-FORMATS&gt;                        Optional, <span class="k">case</span>-insensitive, comma separated list of output
                                           formats. Supported formats are TSV, XML, JSON, GFF3, HTML and
                                           SVG. Default <span class="k">for</span> protein sequences are TSV, XML and
                                           GFF3, or <span class="k">for</span> nucleotide sequences GFF3 and XML.

-g                                         Optional, switch on lookup of corresponding Gene Ontology
                                           annotation <span class="o">(</span>IMPLIES -l lookup option<span class="o">)</span>

-h                                         Optional, display <span class="nb">help</span> information

-i &lt;INPUT-FILE-PATH&gt;                       Optional, path to fasta file that should be loaded on
                                           Master startup. Alternatively, <span class="k">in</span> CONVERT mode, the
                                           InterProScan <span class="m">5</span> XML file to convert.

-l                                         Also include lookup of corresponding InterPro
                                           annotation <span class="k">in</span> the TSV and GFF3 output formats.

-m &lt;MINIMUM-SIZE&gt;                          Optional, minimum nucleotide size of ORF to report. Will
                                           only be considered <span class="k">if</span> n is specified as a sequence type.
                                           Please be aware of the fact that <span class="k">if</span> you specify a too
                                           short value it might be that the analysis takes a very long
                                           time!

-o &lt;EXPLICIT_OUTPUT_FILENAME&gt;              Optional explicit output file name <span class="o">(</span>relative or absolute
                                           path<span class="o">)</span>.  Note that this option, the output directory -d option
                                           and the output file basename -b option are mutually
                                           exclusive. If this option is given, you MUST specify a
                                           single output format using the -f option.  The output file
                                           name will not be modified. Note that specifying an output
                                           file name using this option OVERWRITES ANY EXISTING FILE.

-p                                         Optional, switch on lookup of corresponding Pathway
                                           annotation <span class="o">(</span>IMPLIES -l lookup option<span class="o">)</span>
-t &lt;SEQUENCE-TYPE&gt;                         Optional, the <span class="nb">type</span> of the input sequences <span class="o">(</span>dna/rna <span class="o">(</span>n<span class="o">)</span>
                                           or protein <span class="o">(</span>p<span class="o">))</span>.  The default sequence <span class="nb">type</span> is protein.

-T &lt;TEMP-DIR&gt;                              Optional, specify temporary file directory <span class="o">(</span>relative or
                                           absolute path<span class="o">)</span>. The default location is temp/.

-v                                         Optional, display version number

-r                                          Optional. <span class="s1">&#39;Mode&#39;</span> required <span class="o">(</span> -r <span class="s1">&#39;cluster&#39;</span><span class="o">)</span> to run <span class="k">in</span> cluster mode. These options
                                           are provided but have not been tested with this wrapper script. For
                                           more information on running InterProScan <span class="k">in</span> cluster mode:
                                           https://github.com/ebi-pf-team/interproscan/wiki/ClusterMode

-R                                          Optional. Clusterrunid <span class="o">(</span>crid<span class="o">)</span> required when using cluster mode.
                                           -R unique_id
</pre></div>
</div>
<p>Available InterProScan analyses:</p>
<ul class="simple">
<li><p>CDD</p></li>
<li><p>COILS</p></li>
<li><p>Gene3D</p></li>
<li><p>HAMAP</p></li>
<li><p>MOBIDB</p></li>
<li><p>PANTHER</p></li>
<li><p>Pfam</p></li>
<li><p>PIRSF</p></li>
<li><p>PRINTS</p></li>
<li><p>PROSITE (Profiles and Patterns)</p></li>
<li><p>SFLD</p></li>
<li><p>SMART (unlicensed components only by default - this analysis has simplified post-processing that includes an E-value filter, however you should not expect it to give the same match output as the fully licensed version of SMART)</p></li>
<li><p>SUPERFAMILY</p></li>
<li><p>NCBIFAM (includes the previous TIGRFAM analysis)</p></li>
</ul>
<blockquote>
<div><p>OPTIONS FOR XML PARSER OUTPUTS</p>
<dl class="option-list">
<dt><kbd><span class="option">-F <var>&lt;IPRS output directory&gt;</var></span></kbd></dt>
<dd><p>This is the output directory from InterProScan.</p>
</dd>
<dt><kbd><span class="option">-D <var>&lt;database&gt;</var></span></kbd></dt>
<dd><p>Supply the database responsible for these annotations.</p>
</dd>
<dt><kbd><span class="option">-x <var>&lt;taxon&gt;</var></span></kbd></dt>
<dd><p>NCBI taxon ID of the ID being annotated</p>
</dd>
<dt><kbd><span class="option">-y <var>&lt;type&gt;</var></span></kbd></dt>
<dd><p>Transcript or protein</p>
</dd>
<dt><kbd><span class="option">-n <var>&lt;biocurator&gt;</var></span></kbd></dt>
<dd><p>Name of the biocurator who made these annotations</p>
</dd>
<dt><kbd><span class="option">-M <var>&lt;mapping file&gt;</var></span></kbd></dt>
<dd><p>Optional. Mapping file.</p>
</dd>
<dt><kbd><span class="option">-B <var>&lt;bad seq file&gt;</var></span></kbd></dt>
<dd><p>Optional. Bad input sequence file.</p>
</dd>
</dl>
</div></blockquote>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../goanna/using_goanna_cmd.html" class="btn btn-neutral float-left" title="GOanna on the Command Line" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="using_iprs_in_cyverse_DE.html" class="btn btn-neutral float-right" title="InterProScan on CyVerse" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Fiona M. McCarthy.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>