<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>InterProScan on the Command Line &mdash; AgBase 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Intro" href="../combine_gafs/intro.html" />
    <link rel="prev" title="InterProScan on CyVerse" href="using_iprs_in_cyverse_DE.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            AgBase
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Agbase Home</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../agbase/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">InterProScan</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_iprs_in_cyverse_DE.html"><strong>InterProScan on CyVerse</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>InterProScan on the Command Line</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-interproscan-data-now-including-panther"><strong>Getting the InterProScan Data (now including PANTHER)</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#container-technologies"><strong>Container Technologies</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-interproscan-using-docker"><strong>Running InterProScan using Docker</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-interproscan-container"><strong>Getting the InterProScan Container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-interproscan-with-data"><strong>Running InterProScan with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#getting-the-help-and-usage-statement"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#example-command"><strong>Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#understanding-your-results"><strong>Understanding Your Results</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#running-interproscan-with-singularity-or-apptainer-on-hpc"><strong>Running InterProScan with Singularity (or Apptainer) on HPC</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id1"><strong>Getting the InterProScan Container</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id3"><strong>Getting the Help and Usage Statement</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#id4"><strong>Running InterProScan with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-slurm-script"><strong>Example SLURM Script</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id6"><strong>Understanding Your Results</strong></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Combine GAFs</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cyverse.html"><strong>Combine GAFs on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cmd.html"><strong>Combine GAFs on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">KOBAS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kobas/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_cmd.html"><strong>KOBAS on the Command Line</strong></a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">AgBase</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active"><strong>InterProScan on the Command Line</strong></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/interproscan/using_iprs_cmd.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="interproscan-on-the-command-line">
<h1><strong>InterProScan on the Command Line</strong><a class="headerlink" href="#interproscan-on-the-command-line" title="Permalink to this headline"></a></h1>
<section id="getting-the-interproscan-data-now-including-panther">
<h2><strong>Getting the InterProScan Data (now including PANTHER)</strong><a class="headerlink" href="#getting-the-interproscan-data-now-including-panther" title="Permalink to this headline"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.63-95.0/alt/interproscan-data-5.63-95.0.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.63-95.0/alt/interproscan-data-5.63-95.0.tar.gz.md5
md5sum -c interproscan-data-5.63-95.0.tar.gz.md
tar -pxvzf interproscan-data-5.63-95.0.tar.gz
</pre></div>
</div>
<div class="admonition-tar-options admonition">
<p class="admonition-title">tar options</p>
<ul class="simple">
<li><p>p = preserve the file permissions</p></li>
<li><p>x = extract files from an archive</p></li>
<li><p>v = verbosely list the files processed</p></li>
<li><p>z = filter the archive through gzip</p></li>
<li><p>f = use archive file</p></li>
</ul>
</div>
</section>
<section id="container-technologies">
<h2><strong>Container Technologies</strong><a class="headerlink" href="#container-technologies" title="Permalink to this headline"></a></h2>
<p>Interproscan is provided as a Docker container.</p>
<p>A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another.</p>
<p>There are two major containerization technologies: <strong>Docker</strong> and <strong>Singularity (also known as Apptainer</strong>.</p>
<p>Docker containers can be run with either technology.</p>
</section>
<section id="running-interproscan-using-docker">
<h2><strong>Running InterProScan using Docker</strong><a class="headerlink" href="#running-interproscan-using-docker" title="Permalink to this headline"></a></h2>
<div class="admonition-about-docker admonition">
<p class="admonition-title">About Docker</p>
<ul class="simple">
<li><p>Docker must be installed on the computer you wish to use for your analysis.</p></li>
<li><p>To run Docker you must have ‘root’ permissions (or use sudo).</p></li>
<li><p>Docker will run all containers as ‘root’. This makes Docker incompatible with HPC systems (see Singularity below).</p></li>
<li><p>Docker can be run on your local computer, a server, a cloud virtual machine etc.</p></li>
<li><p>For more information on installing Docker on other systems see this tutorial:  <a class="reference external" href="https://docs.docker.com/engine/install/">Installing Docker on your machine</a>.</p></li>
</ul>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>We have included this basic documentation for running InterProScan with Docker. However, InterProScan requires quite a lot of compute resources and may need to be run on an HPC system. If you need to use HPC see ‘Singularity’ below.</p>
</div>
<section id="getting-the-interproscan-container">
<h3><strong>Getting the InterProScan Container</strong><a class="headerlink" href="#getting-the-interproscan-container" title="Permalink to this headline"></a></h3>
<p>The InterProScan tool is available as a Docker container on Docker Hub where you can see all the available versions:
<a class="reference external" href="https://hub.docker.com/r/agbase/interproscan">InterProScan container</a></p>
<p>The latest container can be pulled with this command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>docker pull agbase/interproscan:5.63-95
</pre></div>
</div>
<div class="admonition-remember admonition">
<p class="admonition-title">Remember</p>
<p>You must have root permissions or use sudo, like so:</p>
<p>sudo docker pull agbase/interproscan:5.63-95</p>
</div>
</section>
<section id="running-interproscan-with-data">
<h3><strong>Running InterProScan with Data</strong><a class="headerlink" href="#running-interproscan-with-data" title="Permalink to this headline"></a></h3>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>There is one directory built into this container. This directory should be used to mount your working directory.</p>
<ul class="simple">
<li><p>/data</p></li>
</ul>
</div>
<section id="getting-the-help-and-usage-statement">
<h4><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#getting-the-help-and-usage-statement" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run --rm -v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir agbase/interproscan:5.63-95 -h
</pre></div>
</div>
<p>See <span class="xref std std-ref">iprsusage</span></p>
</section>
<section id="example-command">
<h4><strong>Example Command</strong><a class="headerlink" href="#example-command" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run <span class="se">\</span>
-v /your/local/data/directory:/data <span class="se">\</span>
-v /where/you/downloaded/interproscan/data/interproscan-5.63-95.0/data:/opt/interproscan/data <span class="se">\</span>
agbase/interproscan:5.63-95 <span class="se">\</span>
-i /path/to/your/input/file/pnnl_10000.fasta <span class="se">\</span>
-d outdir_10000 <span class="se">\</span>
-f tsv,json,xml,gff3 <span class="se">\</span>
-g <span class="se">\</span>
-p <span class="se">\</span>
-c <span class="se">\</span>
-n curator <span class="se">\</span>
-x <span class="m">109069</span> <span class="se">\</span>
-D database <span class="se">\</span>
-l
</pre></div>
</div>
<section id="command-explained">
<h5><strong>Command Explained</strong><a class="headerlink" href="#command-explained" title="Permalink to this headline"></a></h5>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>–rm:</strong> removes container when analysis finishes (image will remain for furture analyses)</p>
<p><strong>-v /your/local/data/directory:/data:</strong> mount my working directory on the host machine into the /data directory in the container. The syntax for this is &lt;absolute path on host machine&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /where/you/downloaded/interproscan/data/interproscan-5.64-95.0/data:/opt/interproscan/data:</strong> mounts the InterProScan partner data (downloaded from FTP) on the host machine into the /opt/interproscan/data directory in the container</p>
<p><strong>agbase/interproscan:5.63-95:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are Interproscan options</p>
</div>
<p><strong>-i /path/to/your/input/file/pnnl_10000.fasta:</strong> local path to input FASTA file. You can also use the mounted file path: /data/pnnl_10000.fasta</p>
<p><strong>-d outdir_10000:</strong> output directory name</p>
<p><strong>-f tsv,json,xml,gff3:</strong> desired output file formats</p>
<p><strong>-g:</strong> tells the tool to perform GO annotation</p>
<p><strong>-p:</strong> tells tool to perform pathway annotaion</p>
<p><strong>-c:</strong> tells tool to perform local compute and not connect to EBI. This only adds a little to the run time but removes error messages from network time out errors</p>
<p><strong>-n curator:</strong> name of biocurator to include in column 15 of GAF output file</p>
<p><strong>-x 109069:</strong> taxon ID of query species to be used in column 13 of GAF output file</p>
<p><strong>-D database:</strong> database of query accession to be used in column 1 of GAF output file</p>
<p><strong>-l:</strong> tells tools to include lookup of corresponding InterPro annotation in the TSV and GFF3 output formats.</p>
</section>
</section>
<section id="understanding-your-results">
<h4><strong>Understanding Your Results</strong><a class="headerlink" href="#understanding-your-results" title="Permalink to this headline"></a></h4>
<section id="interproscan-outputs-https-github-com-ebi-pf-team-interproscan-wiki-outputformats">
<h5><strong>InterProScan outputs:</strong> <a class="reference external" href="https://github.com/ebi-pf-team/interproscan/wiki/OutputFormats">https://github.com/ebi-pf-team/interproscan/wiki/OutputFormats</a><a class="headerlink" href="#interproscan-outputs-https-github-com-ebi-pf-team-interproscan-wiki-outputformats" title="Permalink to this headline"></a></h5>
<ul class="simple">
<li><p>&lt;basename&gt;.gff3</p></li>
<li><p>&lt;basename&gt;.tsv</p></li>
<li><p>&lt;basename&gt;.xml</p></li>
<li><p>&lt;basename&gt;.json</p></li>
</ul>
</section>
<section id="parser-outputs">
<h5><strong>Parser Outputs</strong><a class="headerlink" href="#parser-outputs" title="Permalink to this headline"></a></h5>
<p><strong>&lt;basename&gt;_gaf.txt:</strong>
-This table follows the formatting of a gene association file (gaf) and can be used in GO enrichment analyses.</p>
<p><strong>&lt;basename&gt;_acc_go_counts.txt:</strong>
-This table includes input accessions, the number of GO IDs assigned to each accession and GO ID names. GO IDs are split into BP (Biological Process), MF (Molecular Function) and CC (Cellular Component).</p>
<p><strong>&lt;basename&gt;_go_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each GO ID so that the user can quickly identify all genes assigned to a particular function.</p>
<p><strong>&lt;basename&gt;_acc_interpro_counts.txt:</strong>
-This table includes input accessions, number of InterPro IDs for each accession, InterPro IDs assigned to each sequence and the InterPro ID name.</p>
<p><strong>&lt;basename&gt;_interpro_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each InterPro ID so that the user can quickly identify all genes with a particular motif.</p>
<p><strong>&lt;basename&gt;_acc_pathway_counts.txt:</strong>
-This table includes input accessions, number of pathway IDs for the accession and the pathway names. Multiple values are separated by a semi-colon.</p>
<p><strong>&lt;basename&gt;_pathway_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each Pathway ID so that the user can quickly identify all genes assigned to a pathway.</p>
<p><strong>&lt;basename&gt;.err:</strong>
-This file will list any sequences that were not able to be analyzed by InterProScan. Examples of sequences that will cause an error are sequences with a large run of Xs.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</section>
</section>
</section>
</section>
<section id="running-interproscan-with-singularity-or-apptainer-on-hpc">
<h2><strong>Running InterProScan with Singularity (or Apptainer) on HPC</strong><a class="headerlink" href="#running-interproscan-with-singularity-or-apptainer-on-hpc" title="Permalink to this headline"></a></h2>
<div class="admonition-about-singularity admonition">
<p class="admonition-title">About Singularity</p>
<ul class="simple">
<li><p>does not require ‘root’ permissions</p></li>
<li><p>runs all containers as the user that is logged into the host machine</p></li>
<li><p>HPC systems are likely to have Singularity (or Apptainer) installed and are unlikely to object if asked to install it (no guarantees).</p></li>
<li><p>can be run on any machine where is is installed</p></li>
<li><p>more information about <a class="reference external" href="https://apptainer.org/user-docs/3.8/">Singularity</a> and <a class="reference external" href="https://apptainer.org/docs/user/latest/">Apptainer</a></p></li>
<li><p>This tool was tested using SingularityCE 3.11.4</p></li>
</ul>
</div>
<div class="admonition-hpc-job-schedulers admonition">
<p class="admonition-title">HPC Job Schedulers</p>
<p>Although Singularity can be installed on any computer this documentation assumes it will be run on an HPC system. The tool was tested on a SLURM system and the job submission scripts below reflect that. Submission scripts will need to be modified for use with other job scheduler systems.</p>
</div>
<section id="id1">
<h3><strong>Getting the InterProScan Container</strong><a class="headerlink" href="#id1" title="Permalink to this headline"></a></h3>
<p>The InterProScan tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/interproscan">InterProScan container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>singularity pull docker://agbase/interproscan:5.63-95
</pre></div>
</div>
<section id="id3">
<h4><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h4>
<p><strong>Example SLURM script:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=jobname</span>
<span class="c1">#SBATCH --ntasks=48</span>
<span class="c1">#SBATCH --nodes=1</span>
<span class="c1">#SBATCH --mem=0</span>
<span class="c1">#SBATCH --time=48:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>


module load singularityCE

singularity run <span class="se">\</span>
interproscan_5.63-95.sif <span class="se">\</span>
-h
</pre></div>
</div>
<p>See <span class="xref std std-ref">iprsusage</span></p>
</section>
</section>
<section id="id4">
<h3><strong>Running InterProScan with Data</strong><a class="headerlink" href="#id4" title="Permalink to this headline"></a></h3>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>There is one directory built into this container. This directory should be used to mount your working directory.</p>
<ul class="simple">
<li><p>/data</p></li>
</ul>
</div>
<section id="example-slurm-script">
<h4><strong>Example SLURM Script</strong><a class="headerlink" href="#example-slurm-script" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=jobname</span>
<span class="c1">#SBATCH --ntasks=48</span>
<span class="c1">#SBATCH --nodes=1</span>
<span class="c1">#SBATCH --mem=0</span>
<span class="c1">#SBATCH --time=48:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>

module load singularityCE

singularity run <span class="se">\</span>
-B /your/local/data/directory:/data <span class="se">\</span>
-B /where/you/downloaded/interproscan/data/interproscan-5.63-85.0/data:/opt/interproscan/data <span class="se">\</span>
interproscan_5.63-95.sif <span class="se">\</span>
-i /your/local/data/directory/pnnl_10000.fasta <span class="se">\</span>
-d outdir_10000 <span class="se">\</span>
-f tsv,json,xml,gff3 <span class="se">\</span>
-g <span class="se">\</span>
-p <span class="se">\</span>
-c <span class="se">\</span>
-n biocurator <span class="se">\</span>
-x <span class="m">109069</span> <span class="se">\</span>
-D database <span class="se">\</span>
-l
</pre></div>
</div>
<section id="id5">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id5" title="Permalink to this headline"></a></h5>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /your/local/data/directory:/data:</strong> mounts my working directory on the host machine into the /data directory in the container the syntax for this is &lt;aboslute path on host machine&gt;:&lt;aboslute path in container&gt;</p>
<p><strong>-B /where/you/downloaded/interproscan/data/interproscan-5.63-95.0/data:/opt/interproscan/data:</strong> mounts he InterProScan data directory that was downloaded from the FTP site into the InterProScan data directory in the container</p>
<p><strong>interproscan_5.63-95.sif:</strong> name of the image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are options for this tool</p>
</div>
<p><strong>-i /your/local/data/directory/pnnl_10000.fasta:</strong> input FASTA file</p>
<p><strong>-d outdir_10000:</strong> output directory name</p>
<p><strong>-f tsv,json,xml,gff3:</strong> desired output file formats</p>
<p><strong>-g:</strong> tells the tool to perform GO annotation</p>
<p><strong>-c:</strong> tells tool to perform local compute and not connect to EBI. This only adds a little to the run time but removes error messages from network time out errors</p>
<p><strong>-p:</strong> tells tool to perform pathway annoation</p>
<p><strong>-n biocurator:</strong> name of biocurator to include in column 15 of GAF output file</p>
<p><strong>-x 109069:</strong> taxon ID of query species to be used in column 13 of GAF output file</p>
<p><strong>-D database:</strong> database of query accession to be used in column 1 of GAF output file</p>
<p><strong>-l:</strong> tells tools to include lookup of corresponding InterPro annotation in the TSV and GFF3 output formats.</p>
</section>
</section>
<section id="id6">
<h4><strong>Understanding Your Results</strong><a class="headerlink" href="#id6" title="Permalink to this headline"></a></h4>
<section id="id7">
<h5><strong>InterProScan outputs:</strong> <a class="reference external" href="https://github.com/ebi-pf-team/interproscan/wiki/OutputFormats">https://github.com/ebi-pf-team/interproscan/wiki/OutputFormats</a><a class="headerlink" href="#id7" title="Permalink to this headline"></a></h5>
<ul class="simple">
<li><p>&lt;basename&gt;.gff3</p></li>
<li><p>&lt;basename&gt;.tsv</p></li>
<li><p>&lt;basename&gt;.xml</p></li>
<li><p>&lt;basename&gt;.json</p></li>
</ul>
</section>
<section id="id8">
<h5><strong>Parser Outputs</strong><a class="headerlink" href="#id8" title="Permalink to this headline"></a></h5>
<p><strong>&lt;basename&gt;_gaf.txt:</strong>
-This table follows the formatting of a gene association file (gaf) and can be used in GO enrichment analyses.</p>
<p><strong>&lt;basename&gt;_acc_go_counts.txt:</strong>
-This table includes input accessions, the number of GO IDs assigned to each accession and GO ID names. GO IDs are split into BP (Biological Process), MF (Molecular Function) and CC (Cellular Component).</p>
<p><strong>&lt;basename&gt;_go_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each GO ID so that the user can quickly identify all genes assigned to a particular function.</p>
<p><strong>&lt;basename&gt;_acc_interpro_counts.txt:</strong>
-This table includes input accessions, number of InterPro IDs for each accession, InterPro IDs assigned to each sequence and the InterPro ID name.</p>
<p><strong>&lt;basename&gt;_interpro_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each InterPro ID so that the user can quickly identify all genes with a particular motif.</p>
<p><strong>&lt;basename&gt;_acc_pathway_counts.txt:</strong>
-This table includes input accessions, number of pathway IDs for the accession and the pathway names. Multiple values are separated by a semi-colon.</p>
<p><strong>&lt;basename&gt;_pathway_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each Pathway ID so that the user can quickly identify all genes assigned to a pathway.</p>
<p><strong>&lt;basename&gt;.err:</strong>
-This file will list any sequences that were not able to be analyzed by InterProScan. Examples of sequences that will cause an error are sequences with a large run of Xs.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</section>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="using_iprs_in_cyverse_DE.html" class="btn btn-neutral float-left" title="InterProScan on CyVerse" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../combine_gafs/intro.html" class="btn btn-neutral float-right" title="Intro" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Fiona M. McCarthy.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>