

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Running InterProScan on the Command Line &mdash; AgBase 1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="AgBase 1.0 documentation" href="../index.html"/>
        <link rel="next" title="&lt;no title&gt;" href="../combine_gafs/combine_gafs.html"/>
        <link rel="prev" title="&lt;no title&gt;" href="using_iprs_in_cyverse_DE.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> AgBase
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Agbase Home</span></p>
<ul class="simple">
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../workflow/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">TransDecoder</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro to GOanna</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>Using GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>Running GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">InterProScan</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro to InterProScan</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Running InterProScan on the Command Line</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-interproscan-data"><strong>Getting the InterProScan Data</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#container-technologies"><strong>Container Technologies</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-goanna-using-docker"><strong>Running GOanna using Docker</strong></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-interproscan-container"><strong>Getting the InterProScan Container</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-help-and-usage-statement"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-goanna-with-data"><strong>Running GOanna with Data</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#understanding-your-results"><strong>Understanding Your Results</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-the-container-with-singularity-hpc"><strong>Running the container with Singularity (HPC)</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1"><strong>Getting the InterProScan Data</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#id2"><strong>Getting the InterProScan Container</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#id4"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-interproscan-with-data"><strong>Running InterProScan with Data</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#id5"><strong>Understanding Your Results</strong></a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Combine GAFs</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">KOBAS</span></p>
<ul class="simple">
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">AgBase</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li><strong>Running InterProScan on the Command Line</strong></li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/interproscan/using_iprs_cmd.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="running-interproscan-on-the-command-line">
<h1><strong>Running InterProScan on the Command Line</strong><a class="headerlink" href="#running-interproscan-on-the-command-line" title="Permalink to this headline">¶</a></h1>
<div class="section" id="getting-the-interproscan-data">
<h2><strong>Getting the InterProScan Data</strong><a class="headerlink" href="#getting-the-interproscan-data" title="Permalink to this headline">¶</a></h2>
<p>Partner data are available from the InterProScan FTP site. These data are available as two separate downloads and can be obtained following these instructions:</p>
<p><strong>1. Partner Data excluding Panther</strong></p>
<div class="highlight-none"><div class="highlight"><pre><span></span>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.36-75.0/alt/interproscan-data-5.36-75.0.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.36-75.0/alt/interproscan-data-5.36-75.0.tar.gz.md5
md5sum -c interproscan-data-5.36-75.0.tar.gz.md
tar -pxvzf interproscan-data-5.36-75.0.tar.gz
</pre></div>
</div>
<div class="admonition-tar-options admonition">
<p class="first admonition-title">tar options</p>
<ul class="last simple">
<li>p = preserve the file permissions</li>
<li>x = extract files from an archive</li>
<li>v = verbosely list the files processed</li>
<li>z = filter the archive through gzip</li>
<li>f = use archive file</li>
</ul>
</div>
<p><strong>2. Getting the Panther data</strong></p>
<div class="highlight-none"><div class="highlight"><pre><span></span>cd interproscan-5.36-75.0/data
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/panther-data-14.1.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/panther-data-14.1.tar.gz.md5
md5sum -c panther-data-14.1.tar.gz.md5
tar -pxvzf panther-data-14.1.tar.gz
</pre></div>
</div>
<p>These data will be unarchived into this directory structure. They will need to be mounted to the container at run time.</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>&lt;current_working_directory&gt;/interproscan-5.36-75.0/data
</pre></div>
</div>
<div class="section" id="container-technologies">
<h3><strong>Container Technologies</strong><a class="headerlink" href="#container-technologies" title="Permalink to this headline">¶</a></h3>
<p>GOanna is provided as a Docker container.</p>
<p>A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another.</p>
<p>There are two major containerization technologies: <strong>Docker</strong> and <strong>Singularity</strong>.</p>
<p>Docker containers can be run with either technology.</p>
</div>
<div class="section" id="running-goanna-using-docker">
<h3><strong>Running GOanna using Docker</strong><a class="headerlink" href="#running-goanna-using-docker" title="Permalink to this headline">¶</a></h3>
<div class="admonition-about-docker admonition">
<p class="first admonition-title">About Docker</p>
<ul class="last simple">
<li>Docker must be installed on the computer you wish to use for your analysis.</li>
<li>To run Docker you must have ‘root’ permissions (or use sudo).</li>
<li>Docker will run all containers as ‘root’. This makes Docker incompatible with HPC systems (see Singularity below).</li>
<li>Docker can be run on your local computer, a server, a cloud virtual machine (such as CyVerse Atmosphere) etc. Docker can be installed quickly on an Atmosphere instance by typing ‘ezd’.</li>
<li>For more information on installing Docker on other systems see this tutorial:  <a class="reference external" href="https://learning.cyverse.org/projects/container_camp_workshop_2019/en/latest/docker/dockerintro.html">Installing Docker on your machine</a>.</li>
</ul>
</div>
<div class="admonition important">
<p class="first admonition-title">Important</p>
<p class="last">We have included this basic documentation for running InterProScan with Docker. However, InterProScan requires quite a lot of compute resources and may need to be run on an HPC system. If you need to use HPC see &#8216;Singularity&#8217; below.</p>
</div>
</div>
</div>
<div class="section" id="getting-the-interproscan-container">
<h2><strong>Getting the InterProScan Container</strong><a class="headerlink" href="#getting-the-interproscan-container" title="Permalink to this headline">¶</a></h2>
<p>The InterProScan tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/interproscan">InterProScan container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker pull agbase/interproscan:5.36-75.0_0
</pre></div>
</div>
<div class="admonition-remember admonition">
<p class="first admonition-title">Remember</p>
<p>You must have root permissions or use sudo, like so:</p>
<p class="last">sudo docker pull agbase/interproscan:5.36-75.0_0</p>
</div>
</div>
<div class="section" id="getting-the-help-and-usage-statement">
<h2><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#getting-the-help-and-usage-statement" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash"><div class="highlight"><pre><span></span>sudo docker run --rm -v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir agbase/interproscan:5.36-75.0_0 -h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>   Options:
-a  &lt;ANALYSES&gt;                             Optional, comma separated list of analyses.  If this option
                                           is not set, ALL analyses will be run.

-b, &lt;OUTPUT-FILE-BASE&gt;                     Optional, base output filename (relative or absolute path).
                                           Note that this option, the output directory (-d) option and
                                           the output file name (-o) option are mutually exclusive.  The
                                           appropriate file extension for the output format(s) will be
                                           appended automatically. By default the input file
                                           path/name will be used.

-d,&lt;OUTPUT-DIR&gt;                            Optional, output directory. Note that this option, the
                                           output file name (-o) option and the output file base (-b) option
                                           are mutually exclusive. The output filename(s) are the
                                           same as the input filename, with the appropriate file
                                           extension(s) for the output format(s) appended automatically .

-c                                         Optional.  Disables use of the precalculated match lookup
                                           service.  All match calculations will be run locally.

-C                                         Optional. Supply the number of cpus to use.

-e                                         Optional, excludes sites from the XML, JSON output

-f &lt;OUTPUT-FORMATS&gt;                        Optional, case-insensitive, comma separated list of output
                                           formats. Supported formats are TSV, XML, JSON, GFF3, HTML and
                                           SVG. Default for protein sequences are TSV, XML and
                                           GFF3, or for nucleotide sequences GFF3 and XML.

-g                                         Optional, switch on lookup of corresponding Gene Ontology
                                           annotation (IMPLIES -l lookup option)

-h                                         Optional, display help information

-i &lt;INPUT-FILE-PATH&gt;                       Optional, path to fasta file that should be loaded on
                                           Master startup. Alternatively, in CONVERT mode, the
                                           InterProScan 5 XML file to convert.

-l                                         Also include lookup of corresponding InterPro
                                           annotation in the TSV and GFF3 output formats.

-m &lt;MINIMUM-SIZE&gt;                          Optional, minimum nucleotide size of ORF to report. Will
                                           only be considered if n is specified as a sequence type.
                                           Please be aware of the fact that if you specify a too
                                           short value it might be that the analysis takes a very long
                                           time!

-o &lt;EXPLICIT_OUTPUT_FILENAME&gt;              Optional explicit output file name (relative or absolute
                                           path).  Note that this option, the output directory -d option
                                           and the output file basename -b option are mutually
                                           exclusive. If this option is given, you MUST specify a
                                           single output format using the -f option.  The output file
                                           name will not be modified. Note that specifying an output
                                           file name using this option OVERWRITES ANY EXISTING FILE.

-p                                         Optional, switch on lookup of corresponding Pathway
                                           annotation (IMPLIES -l lookup option)
-t &lt;SEQUENCE-TYPE&gt;                         Optional, the type of the input sequences (dna/rna (n)
                                           or protein (p)).  The default sequence type is protein.

-T &lt;TEMP-DIR&gt;                              Optional, specify temporary file directory (relative or
                                           absolute path). The default location is temp/.

-v                                         Optional, display version number

-r                                          Optional. &#39;Mode&#39; required ( -r &#39;cluster&#39;) to run in cluster mode. These options
                                           are provided but have not been tested with this wrapper script. For
                                           more information on running InterProScan in cluster mode:
                                           https://github.com/ebi-pf-team/interproscan/wiki/ClusterMode

-R                                          Optional. Clusterrunid (crid) required when using cluster mode.
                                           -R unique_id
Available analyses:
                     TIGRFAM (XX.X) : TIGRFAMs are protein families based on Hidden Markov Models or HMMs
                        SFLD (X.X) : SFLDs are protein families based on Hidden Markov Models or HMMs
                       ProDom (XXXX.X) : ProDom is a comprehensive set of protein domain families automatically generated from the UniProt Knowledge Database.
                       Hamap (XXXXXX.XX) : High-quality Automated and Manual Annotation of Microbial Proteomes
                       SMART (X.X) : SMART allows the identification and analysis of domain architectures based on Hidden Markov Models or HMMs
                         CDD (X.XX) : Prediction of CDD domains in Proteins
             ProSiteProfiles (XX.XXX) : PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them
             ProSitePatterns (XX.XXX) : PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them
                 SUPERFAMILY (X.XX) : SUPERFAMILY is a database of structural and functional annotation for all proteins and genomes.
                      PRINTS (XX.X) : A fingerprint is a group of conserved motifs used to characterise a protein family
                     PANTHER (X.X) : The PANTHER (Protein ANalysis THrough Evolutionary Relationships) Classification System is a unique resource that classifies genes by their functions, using published scientific experimental evidence and evolutionary relationships to predict fu$
                      Gene3D (X.X.X) : Structural assignment for whole genes and genomes using the CATH domain structure database
                       PIRSF (X.XX) : The PIRSF concept is being used as a guiding principle to provide comprehensive and non-overlapping clustering of UniProtKB sequences into a hierarchical order to reflect their evolutionary relationships.
                        Pfam (XX.X) : A large collection of protein families, each represented by multiple sequence alignments and hidden Markov models (HMMs)
                       Coils (X.X) : Prediction of Coiled Coil Regions in Proteins
                  MobiDBLite (X.X) : Prediction of disordered domains Regions in Proteins

OPTIONS FOR XML PARSER OUTPUTS

-F &lt;IPRS output directory&gt;              This is the output directory from InterProScan.
-D &lt;database&gt;                           Supply the database responsible for these annotations.
-x &lt;taxon&gt;                              NCBI taxon ID of the ID being annotated
-y &lt;type&gt;                               Transcript or protein
-n &lt;biocurator&gt;                         Name of the biocurator who made these annotations
-M &lt;mapping file&gt;                       Optional. Mapping file.
-B &lt;bad seq file&gt;                       Optional. Bad input sequence file.
</pre></div>
</div>
</div>
<div class="section" id="running-goanna-with-data">
<h2><strong>Running GOanna with Data</strong><a class="headerlink" href="#running-goanna-with-data" title="Permalink to this headline">¶</a></h2>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There are one directory built into this container. This directory should be used to mount your working directory.</p>
<ul class="last simple">
<li>/data</li>
</ul>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>sudo docker run \
-v /rsgrps/shaneburgess/amanda/i5k/interproscan:/data \
-v /rsgrps/shaneburgess/amanda/i5k/interproscan/interproscan-5.36-75.0/data:/opt/interproscan/data \
agbase/interproscan:5.36-75.0-0 \
-i /rsgrps/shaneburgess/amanda/i5k/interproscan/pnnl_10000.fasta \
-d outdir_10000 \
-f tsv,json,xml,html,gff3,svg \
-g \
-p \
-n Amanda \
-x 109069 \
-D AgBase
</pre></div>
</div>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>&#8211;rm:</strong> removes container when analysis finishes (image will remain for furture analyses)</p>
<p><strong>-v /rsgrps/shaneburgess/amanda/i5k/interproscan:/data:</strong> mount my working directory on the host machine into the /data directory in the container. The syntax for this is &lt;absolute path on host machine&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /rsgrps/shaneburgess/amanda/i5k/interproscan/interproscan-5.36-75.0/data:/opt/interproscan/data:</strong> mounts the InterProScan partner data (downloaded from FTP) on the host machine into the /opt/interproscan/data directory in the container</p>
<p><strong>agbase/interproscan:5.36-75.0-0:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are GOanna options</p>
</div>
<p><strong>-i /rsgrps/shaneburgess/amanda/i5k/interproscan/pnnl_10000.fasta:</strong> input FASTA file</p>
<p><strong>-d outdir_10000:</strong> output directory name</p>
<p><strong>-f tsv,json,xml,html,gff3,svg:</strong> desired output file formats</p>
<p><strong>-g:</strong> tells the tool to perform GO annotation</p>
<p><strong>-p:</strong> tells tool to perform pathway annoation</p>
<p><strong>-n Amanda:</strong> name of biocurator to include in column 15 of GAF output file</p>
<p><strong>-x 109069:</strong> taxon ID of query species to be used in column 13 of GAF output file</p>
<p><strong>-D AgBase:</strong> database of query accession to be used in column 1 of GAF output file</p>
</div>
<div class="section" id="understanding-your-results">
<h2><strong>Understanding Your Results</strong><a class="headerlink" href="#understanding-your-results" title="Permalink to this headline">¶</a></h2>
<p><strong>Default</strong>
- &lt;basename&gt;.gff3
- &lt;basename&gt;.tsv
- &lt;basename&gt;.xml
<strong>Optional</strong>
- &lt;basename&gt;.json
- &lt;basename&gt;.html.tar.gz
- &lt;basename&gt;.svg.tar.gz</p>
<p><strong>&lt;basename&gt;_gaf.txt:</strong>
-This table follows the formatting of a gene association file (gaf) and can be used in GO enrichment analyses.</p>
<p><strong>&lt;basename&gt;_acc_go_counts.txt:</strong>
-This table includes input accessions, the number of GO IDs assigned to each accession and GO ID names. GO IDs are split into BP (Biological Process), MF (Molecular Function) and CC (Cellular Component).</p>
<p><strong>&lt;basename&gt;_go_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each GO ID so that the user can quickly identify all genes assigned to a particular function.</p>
<p><strong>&lt;basename&gt;_acc_interpro_counts.txt:</strong>
-This table includes input accessions, number of InterPro IDs for each accession, InterPro IDs assigned to each sequence and the InterPro ID name.</p>
<p><strong>&lt;basename&gt;_interpro_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each InterPro ID so that the user can quickly identify all genes with a particular motif.</p>
<p><strong>&lt;basename&gt;_acc_pathway_counts.txt:</strong>
-This table includes input accessions, number of pathway IDs for the accession and the pathway names. Multiple values are separated by a semi-colon.</p>
<p><strong>&lt;basename&gt;_pathway_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each Pathway ID so that the user can quickly identify all genes assigned to a pathway.</p>
<p><strong>&lt;basename&gt;.err:</strong>
-This file will list any sequences that were not able to be analyzed by InterProScan. Examples of sequences that will cause an error are sequences with a large run of Xs.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</div>
<div class="section" id="running-the-container-with-singularity-hpc">
<h2><strong>Running the container with Singularity (HPC)</strong><a class="headerlink" href="#running-the-container-with-singularity-hpc" title="Permalink to this headline">¶</a></h2>
<div class="admonition-about-singularity admonition">
<p class="first admonition-title">About Singularity</p>
<ul class="last simple">
<li>does not require ‘root’ permissions</li>
<li>runs all containers as the user that is logged into the host machine</li>
<li>HPC systems are likely to have Singularity installed and are unlikely to object if asked to install it (no guarantees).</li>
<li>can be run on any machine where is is installed</li>
<li>more information about <a class="reference external" href="https://singularity.lbl.gov/docs-installation">installing Singularity</a></li>
<li>This tool was tested using Singularity 3.0. Users with Singularity 2.x will need to modify the commands accordingly.</li>
</ul>
</div>
<div class="admonition-hpc-job-schedulers admonition">
<p class="first admonition-title">HPC Job Schedulers</p>
<p class="last">Although Singularity can be installed on any computer this documentation assumes it will be run on an HPC system. The tool was tested on a PBSPro system and the job submission scripts below reflect that. Submission scripts will need to be modified for use with other job scheduler systems.</p>
</div>
</div>
<div class="section" id="id1">
<h2><strong>Getting the InterProScan Data</strong><a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>Partner data are available from the InterProScan FTP site. These data are available as two separate downloads and can be obtained following these instructions:</p>
<p><strong>1. Partner Data excluding Panther</strong></p>
<div class="highlight-none"><div class="highlight"><pre><span></span>wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.36-75.0/alt/interproscan-data-5.36-75.0.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.36-75.0/alt/interproscan-data-5.36-75.0.tar.gz.md5
md5sum -c interproscan-data-5.36-75.0.tar.gz.md
tar -pxvzf interproscan-data-5.36-75.0.tar.gz
</pre></div>
</div>
<div class="admonition-tar-options admonition">
<p class="first admonition-title">tar options</p>
<ul class="last simple">
<li>p = preserve the file permissions</li>
<li>x = extract files from an archive</li>
<li>v = verbosely list the files processed</li>
<li>z = filter the archive through gzip</li>
<li>f = use archive file</li>
</ul>
</div>
<p><strong>2. Getting the Panther data</strong></p>
<div class="highlight-none"><div class="highlight"><pre><span></span>cd interproscan-5.36-75.0/data
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/panther-data-14.1.tar.gz
wget ftp://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/data/panther-data-14.1.tar.gz.md5
md5sum -c panther-data-14.1.tar.gz.md5
tar -pxvzf panther-data-14.1.tar.gz
</pre></div>
</div>
<p>These data will be unarchived into this directory structure. They will need to be mounted to the container at run time.</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>&lt;current_working_directory&gt;/interproscan-5.36-75.0/data
</pre></div>
</div>
</div>
<div class="section" id="id2">
<h2><strong>Getting the InterProScan Container</strong><a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h2>
<p>The InterProScan tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/interproscan">InterProScan container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>singularity pull docker://agbase/interproscan:5.36-75.0_0
</pre></div>
</div>
</div>
<div class="section" id="id4">
<h2><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N 10000j100</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/interproscan

singularity pull docker://agbase/interproscan:5.36-75.0_0

singularity run <span class="se">\</span>
interproscan_5.36-75.0_0.sif <span class="se">\</span>
-h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>**INSERT HELP STATEMENT HERE**
</pre></div>
</div>
</div>
<div class="section" id="running-interproscan-with-data">
<h2><strong>Running InterProScan with Data</strong><a class="headerlink" href="#running-interproscan-with-data" title="Permalink to this headline">¶</a></h2>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There is one directory built into this container. This directory should be used to mount your working directory.</p>
<ul class="last simple">
<li>/data</li>
</ul>
</div>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N 10000j100</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/interproscan

singularity pull docker://agbase/interproscan:5.36-75.0_0

singularity run <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/interproscan:/data <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/interproscan/interproscan-5.36-75.0/data:/opt/interproscan/data <span class="se">\</span>
interproscan_5.36-75.0_0.sif <span class="se">\</span>
-i /rsgrps/shaneburgess/amanda/i5k/interproscan/pnnl_10000.fasta <span class="se">\</span>
-d outdir_10000 <span class="se">\</span>
-f tsv,json,xml,html,gff3,svg <span class="se">\</span>
-g <span class="se">\</span>
-p <span class="se">\</span>
-n Amanda <span class="se">\</span>
-x <span class="m">109069</span> <span class="se">\</span>
-D AgBase
</pre></div>
</div>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/interproscan:/data:</strong> mounts my working directory on the host machine into the /data directory in the container the syntax for this is &lt;aboslute path on host machine&gt;:&lt;aboslute path in container&gt;</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/interproscan/interproscan-5.36-75.0/data:/opt/interproscan/data:</strong> mounts he InterProScan data directory that was downloaded from the FTP site into the InterProScan data directory in the container</p>
<p><strong>interproscan_5.36-75.0_0.sif:</strong> name of the image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are options for this tool</p>
</div>
<p><strong>-i /rsgrps/shaneburgess/amanda/i5k/interproscan/pnnl_10000.fasta:</strong> input FASTA file</p>
<p><strong>-d outdir_10000:</strong> output directory name</p>
<p><strong>-f tsv,json,xml,html,gff3,svg:</strong> desired output file formats</p>
<p><strong>-g:</strong> tells the tool to perform GO annotation</p>
<p><strong>-p:</strong> tells tool to perform pathway annoation</p>
<p><strong>-n Amanda:</strong> name of biocurator to include in column 15 of GAF output file</p>
<p><strong>-x 109069:</strong> taxon ID of query species to be used in column 13 of GAF output file</p>
<p><strong>-D AgBase:</strong> database of query accession to be used in column 1 of GAF output file</p>
</div>
<div class="section" id="id5">
<h2><strong>Understanding Your Results</strong><a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h2>
<p><strong>Default</strong>
- &lt;basename&gt;.gff3
- &lt;basename&gt;.tsv
- &lt;basename&gt;.xml
<strong>Optional</strong>
- &lt;basename&gt;.json
- &lt;basename&gt;.html.tar.gz
- &lt;basename&gt;.svg.tar.gz</p>
<p><strong>&lt;basename&gt;_gaf.txt:</strong>
-This table follows the formatting of a gene association file (gaf) and can be used in GO enrichment analyses.</p>
<p><strong>&lt;basename&gt;_acc_go_counts.txt:</strong>
-This table includes input accessions, the number of GO IDs assigned to each accession and GO ID names. GO IDs are split into BP (Biological Process), MF (Molecular Function) and CC (Cellular Component).</p>
<p><strong>&lt;basename&gt;_go_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each GO ID so that the user can quickly identify all genes assigned to a particular function.</p>
<p><strong>&lt;basename&gt;_acc_interpro_counts.txt:</strong>
-This table includes input accessions, number of InterPro IDs for each accession, InterPro IDs assigned to each sequence and the InterPro ID name.</p>
<p><strong>&lt;basename&gt;_interpro_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each InterPro ID so that the user can quickly identify all genes with a particular motif.</p>
<p><strong>&lt;basename&gt;_acc_pathway_counts.txt:</strong>
-This table includes input accessions, number of pathway IDs for the accession and the pathway names. Multiple values are separated by a semi-colon.</p>
<p><strong>&lt;basename&gt;_pathway_counts.txt:</strong>
-This table counts the numbers of sequences assigned to each Pathway ID so that the user can quickly identify all genes assigned to a pathway.</p>
<p><strong>&lt;basename&gt;.err:</strong>
-This file will list any sequences that were not able to be analyzed by InterProScan. Examples of sequences that will cause an error are sequences with a large run of Xs.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../combine_gafs/combine_gafs.html" class="btn btn-neutral float-right" title="&lt;no title&gt;" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="using_iprs_in_cyverse_DE.html" class="btn btn-neutral" title="&lt;no title&gt;" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Fiona M. McCarthy.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>