

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>KOBAS on the Command Line &mdash; AgBase 1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="AgBase 1.0 documentation" href="../index.html"/>
        <link rel="prev" title="KOBAS on CyVerse" href="using_kobas_in_cyverse_DE.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> AgBase
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Agbase Home</span></p>
<ul class="simple">
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../workflow/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">TransDecoder</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">InterProScan</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">Combine GAFs</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">KOBAS</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>KOBAS on the Command Line</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-databases"><strong>Getting the databases</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#container-technologies"><strong>Container Technologies</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-kobas-using-docker"><strong>Running KOBAS using Docker</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-kobas-container"><strong>Getting the KOBAS container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-help-and-usage-statement"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-kobas-with-data"><strong>Running KOBAS with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#annotate-example-command"><strong>Annotate Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-annotate-command"><strong>Breakdown of Annotate Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#identify-example-command"><strong>Identify Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-identify-command"><strong>Breakdown of Identify Command</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#running-kobas-annoate-and-identify-as-a-pipeline"><strong>Running KOBAS Annoate and Identify as a Pipeline</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#annotate-and-identify-pipeline-example-command"><strong>Annotate and Identify Pipeline Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-annotate-and-identify-pipeline-command"><strong>Breakdown of Annotate and Identify Pipeline Command</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#running-kobas-annotate-using-singularity"><strong>Running KOBAS Annotate using Singularity</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id1"><strong>Getting the KOBAS container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id4"><strong>Running KOBAS with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-pbs-script-for-annotate-process"><strong>Example PBS Script for Annotate Process</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-command"><strong>Breakdown of Command</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#running-kobas-identify-using-singularity"><strong>Running KOBAS Identify using Singularity</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id5"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-kobas-identify-with-data"><strong>Running KOBAS Identify with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-pbs-script-for-identify-process"><strong>Example PBS Script for Identify Process</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id6"><strong>Breakdown of Identify Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#example-pbs-script-for-annotate-and-identify-pipeline"><strong>Example PBS Script for Annotate and Identify Pipeline</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-annotate-and-idenfity-command"><strong>Breakdown of Annotate and Idenfity Command</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#understanding-your-results"><strong>Understanding Your Results</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#annotate"><strong>Annotate</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#identifyresults"><strong>Identify</strong></a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">AgBase</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li><strong>KOBAS on the Command Line</strong></li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/kobas/using_kobas_cmd.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="kobas-on-the-command-line">
<h1><strong>KOBAS on the Command Line</strong><a class="headerlink" href="#kobas-on-the-command-line" title="Permalink to this headline">¶</a></h1>
<div class="section" id="getting-the-databases">
<h2><strong>Getting the databases</strong><a class="headerlink" href="#getting-the-databases" title="Permalink to this headline">¶</a></h2>
<p>To run the tool you need some public data. The files can be downloaded directly fro the <a class="reference external" href="kobas.cbi.pku.edu.cn">KOBAS homepage</a>. These directories are also available as two tar archives in the CyVerse Data Store. The files are best downloaded with <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html">iCommands</a>. Once iCommands is <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html#icommands-first-time-configuration">setup</a> you can use ‘iget’ to download the data.</p>
<ol class="arabic simple">
<li>seq_pep.tar: species-specific BLAST databases used by KOBAS</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>iget /iplant/home/shared/iplantcollaborative/protein_blast_dbs/kobas/seq_pep.tar

tar -xf seq_pep.tar
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li>sqlite3.tar: species-specific annotation databases used by KOBAS</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>iget /iplant/home/shared/iplantcollaborative/protein_blast_dbs/kobas/sqlite3.tar

tar -xf sqlite3.tar
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The above commands should result in two directories (seq_pep and sqlite3) each containing many files. There is no need to unzip the .gz files.</p>
</div>
</div>
<div class="section" id="container-technologies">
<h2><strong>Container Technologies</strong><a class="headerlink" href="#container-technologies" title="Permalink to this headline">¶</a></h2>
<p>KOBAS is provided as a Docker container.</p>
<p>A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another.</p>
<p>There are two major containerization technologies: <strong>Docker</strong> and <strong>Singularity</strong>.</p>
<p>Docker containers can be run with either technology.</p>
</div>
<div class="section" id="running-kobas-using-docker">
<h2><strong>Running KOBAS using Docker</strong><a class="headerlink" href="#running-kobas-using-docker" title="Permalink to this headline">¶</a></h2>
<div class="admonition-about-docker admonition">
<p class="first admonition-title">About Docker</p>
<ul class="last simple">
<li>Docker must be installed on the computer you wish to use for your analysis.</li>
<li>To run Docker you must have ‘root’ permissions (or use sudo).</li>
<li>Docker will run all containers as ‘root’. This makes Docker incompatible with HPC systems (see Singularity below).</li>
<li>Docker can be run on your local computer, a server, a cloud virtual machine (such as CyVerse Atmosphere) etc. Docker can be installed quickly on an Atmosphere instance by typing ‘ezd’.</li>
<li>For more information on installing Docker on other systems see this tutorial:  <a class="reference external" href="https://learning.cyverse.org/projects/container_camp_workshop_2019/en/latest/docker/dockerintro.html">Installing Docker on your machine</a>.</li>
</ul>
</div>
<div class="section" id="getting-the-kobas-container">
<h3><strong>Getting the KOBAS container</strong><a class="headerlink" href="#getting-the-kobas-container" title="Permalink to this headline">¶</a></h3>
<p>The KOBAS tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/kobas">KOBAS container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker pull agbase/kobas:3.0.3_0
</pre></div>
</div>
<div class="admonition-remember admonition">
<p class="first admonition-title">Remember</p>
<p>You must have root permissions or use sudo, like so:</p>
<p class="last">sudo docker pull agbase/kobas:3.0.3_0</p>
</div>
</div>
<div class="section" id="getting-the-help-and-usage-statement">
<h3><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#getting-the-help-and-usage-statement" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>sudo docker run --rm -v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir agbase/kobas:3.0.3_0 -h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>Options:
[-h prints this help statement]

[-a runs KOBAS annotate]
KOBAS annotate options:
    -i INFILE can be FASTA or one-per-lineidentifiers. See -t intype for details.
    -s SPECIES 3 or 4 letter species abbreviation (can be found here: ftp://ftp.cbi.pku.edu.cn/pub/KOBAS_3.0_DOWNLOAD/species_abbr.txt or here: https://www.kegg.jp/kegg/catalog/org_list.html)
    -o OUTPUT file (Default is stdout.)
    [-l LIST available species, or list available databases for a specific species]
    [-t INTYPE (fasta:pro, fasta:nuc, blastout:xml, blastout:tab, id:ncbigi, id:uniprot, id:ensembl, id:ncbigene), default fasta:pro]
    [-e EVALUE expect threshold for BLAST, default 1e-5]
    [-r RANK rank cutoff for valid hits from BLAST result, default is 5]
    [-C COVERAGE subject coverage cutoff for BLAST, default 0]
    [-z ORTHOLOG whether only use orthologs for cross-species annotation or not, default NO (if only using orthologs, please provide the species abbreviation of your input)]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/, This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]
    [-T number of THREADS to use in BLAST search. Default = 8]

[-g runs KOBAS identify]
    KOBAS identify options:
    -f FGFILE foreground file, the output of annotate
    -b BGFILE background file, species abbreviation, see this list for species codes: https://www.kegg.jp/kegg/catalog/org_list.html
    -o OUTPUT file (Default is stdout.)
    [-d DB databases for selection, 1-letter abbreviation separated by &quot;/&quot;: K for KEGG PATHWAY, n for PID, b for BioCarta, R for Reactome, B for BioCyc, p for PANTHER,
           o for OMIM, k for KEGG DISEASE, f for FunDO, g for GAD, N for NHGRI GWAS Catalog and G for Gene Ontology, default K/n/b/R/B/p/o/k/f/g/N/]
    [-m METHOD choose statistical test method: b for binomial test, c for chi-square test, h for hypergeometric test / Fisher&#39;s exact test, and x for frequency list,
           default hypergeometric test / Fisher&#39;s exact test
    [-n FDR choose false discovery rate (FDR) correction method: BH for Benjamini and Hochberg, BY for Benjamini and Yekutieli, QVALUE, and None, default BH
    [-c CUTOFF terms with less than cutoff number of genes are not used for statistical tests, default 5]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/. This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]

[-j runs both KOBAS annotate and identify]
</pre></div>
</div>
</div>
<div class="section" id="running-kobas-with-data">
<h3><strong>Running KOBAS with Data</strong><a class="headerlink" href="#running-kobas-with-data" title="Permalink to this headline">¶</a></h3>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<dl class="last docutils">
<dt>There are 3 directories built into this container. These directories should be used to mount data.</dt>
<dd><ul class="first last simple">
<li>/work-dir</li>
<li>/seq_pep</li>
<li>/sqlite3</li>
</ul>
</dd>
</dl>
</div>
<p>KOBAS can perform two tasks:
- annotate (-a)
- identify (enrichment) (-g)
KOBAS can also run both task with a single command (-j).</p>
<div class="section" id="annotate-example-command">
<h4><strong>Annotate Example Command</strong><a class="headerlink" href="#annotate-example-command" title="Permalink to this headline">¶</a></h4>
<div class="highlight-none"><div class="highlight"><pre><span></span>sudo docker run \
--rm \
-v /home/amcooksey/i5k/seq_pep:/seq_pep \
-v /home/amcooksey/i5k/sqlite3:/sqlite3 \
-v $(pwd):/work-dir \
agbase/kobas:3.0.3_0 \
-a
-i AROS1000.fa \
-s dme \
-t fasta:pro
-o AROS1000
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-annotate-command">
<h4><strong>Breakdown of Annotate Command</strong><a class="headerlink" href="#breakdown-of-annotate-command" title="Permalink to this headline">¶</a></h4>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>&#8211;rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v /home/amcooksey/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /home/amcooksey/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>agbase/kobas:3.0.3_0:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-a:</strong> Tells KOBAS to runt he &#8216;annotate&#8217; process.</p>
<p><strong>-i AROS1000.fa:</strong> input file (peptide FASTA)</p>
<p><strong>-s dme:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o AROS1000:</strong> name of output file</p>
<p>For information on output files see <a class="reference internal" href="#annotateresults">annotateresults</a></p>
</div>
<div class="section" id="identify-example-command">
<h4><strong>Identify Example Command</strong><a class="headerlink" href="#identify-example-command" title="Permalink to this headline">¶</a></h4>
<div class="highlight-none"><div class="highlight"><pre><span></span>sudo docker run \
--rm \
-v /home/amcooksey/i5k/seq_pep:/seq_pep \
-v /home/amcooksey/i5k/sqlite3:/sqlite3 \
-v $(pwd):/work-dir \
agbase/kobas:3.0.3_0 \
-g \
-f AROS1000 \
-b dme \
-o ident_out
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-identify-command">
<h4><strong>Breakdown of Identify Command</strong><a class="headerlink" href="#breakdown-of-identify-command" title="Permalink to this headline">¶</a></h4>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>&#8211;rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v /home/amcooksey/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /home/amcooksey/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>agbase/kobas:3.0.3_0:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-g:</strong> Tells KOBAS to runt he &#8216;identify&#8217; process.</p>
<p><strong>-f AROS1000:</strong> output file from KOBAS annotate</p>
<p><strong>-b dme:</strong> background; enter the species code for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-o ident_out:</strong> name of output file</p>
<p>For information of outputs see <a class="reference internal" href="#identifyresults">identifyresults</a></p>
</div>
</div>
<div class="section" id="running-kobas-annoate-and-identify-as-a-pipeline">
<h3><strong>Running KOBAS Annoate and Identify as a Pipeline</strong><a class="headerlink" href="#running-kobas-annoate-and-identify-as-a-pipeline" title="Permalink to this headline">¶</a></h3>
<div class="section" id="annotate-and-identify-pipeline-example-command">
<h4><strong>Annotate and Identify Pipeline Example Command</strong><a class="headerlink" href="#annotate-and-identify-pipeline-example-command" title="Permalink to this headline">¶</a></h4>
<div class="highlight-none"><div class="highlight"><pre><span></span>sudo docker run \
--rm \
-v /home/amcooksey/i5k/seq_pep:/seq_pep \
-v /home/amcooksey/i5k/sqlite3:/sqlite3 \
-v $(pwd):/work-dir \
agbase/kobas:3.0.3_0 \
-j
-i AROS1000.fa \
-s dme \
-t fasta:pro
-o AROS1000
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-annotate-and-identify-pipeline-command">
<h4><strong>Breakdown of Annotate and Identify Pipeline Command</strong><a class="headerlink" href="#breakdown-of-annotate-and-identify-pipeline-command" title="Permalink to this headline">¶</a></h4>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>&#8211;rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v /home/amcooksey/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /home/amcooksey/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>agbase/kobas:3.0.3_0:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-j:</strong> Tells KOBAS to runt he &#8216;annotate&#8217; process.</p>
<p><strong>-i AROS1000.fa:</strong> input file (peptide FASTA)</p>
<p><strong>-s dme:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o AROS1000:</strong> basename of output files</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This pipeline will automatically use the output of &#8216;annotate&#8217; as the -f foreground input for &#8216;identify.
This will also use your species option as the -b background input for &#8216;identify&#8217;.</p>
</div>
<p>For more information on outputs see <a class="reference internal" href="#annoident">annoident</a></p>
</div>
</div>
<div class="section" id="running-kobas-annotate-using-singularity">
<h3><strong>Running KOBAS Annotate using Singularity</strong><a class="headerlink" href="#running-kobas-annotate-using-singularity" title="Permalink to this headline">¶</a></h3>
<div class="admonition-about-singularity admonition">
<p class="first admonition-title">About Singularity</p>
<ul class="last simple">
<li>does not require ‘root’ permissions</li>
<li>runs all containers as the user that is logged into the host machine</li>
<li>HPC systems are likely to have Singularity installed and are unlikely to object if asked to install it (no guarantees).</li>
<li>can be run on any machine where is is installed</li>
<li>more information about <a class="reference external" href="https://singularity.lbl.gov/docs-installation">installing Singularity</a></li>
<li>This tool was tested using Singularity 3.0. Users with Singularity 2.x will need to modify the commands accordingly.</li>
</ul>
</div>
<div class="admonition-hpc-job-schedulers admonition">
<p class="first admonition-title">HPC Job Schedulers</p>
<p class="last">Although Singularity can be installed on any computer this documentation assumes it will be run on an HPC system. The tool was tested on a PBSPro system and the job submission scripts below reflect that. Submission scripts will need to be modified for use with other job scheduler systems.</p>
</div>
</div>
<div class="section" id="id1">
<h3><strong>Getting the KOBAS container</strong><a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>The KOBAS tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/kobas">KOBAS container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>singularity pull docker://agbase/kobas:3.0.3_0
</pre></div>
</div>
</div>
<div class="section" id="id3">
<h3><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p><strong>Example PBS script:</strong></p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N kobas</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/kobas

singularity pull docker://agbase/kobas:3.0.3_0

singularity run <span class="se">\</span>
kobas_3.0.3_0.sif <span class="se">\</span>
-h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>Options:
[-h prints this help statement]

[-a runs KOBAS annotate]
KOBAS annotate options:
    -i INFILE can be FASTA or one-per-lineidentifiers. See -t intype for details.
    -s SPECIES 3 or 4 letter species abbreviation (can be found here: ftp://ftp.cbi.pku.edu.cn/pub/KOBAS_3.0_DOWNLOAD/species_abbr.txt or here: https://www.kegg.jp/kegg/catalog/org_list.html)
    -o OUTPUT file (Default is stdout.)
    -t INTYPE (fasta:pro, fasta:nuc, blastout:xml, blastout:tab, id:ncbigi, id:uniprot, id:ensembl, id:ncbigene), default fasta:pro
    [-l LIST available species, or list available databases for a specific species]
    [-e EVALUE expect threshold for BLAST, default 1e-5]
    [-r RANK rank cutoff for valid hits from BLAST result, default is 5]
    [-C COVERAGE subject coverage cutoff for BLAST, default 0]
    [-z ORTHOLOG whether only use orthologs for cross-species annotation or not, default NO (if only using orthologs, please provide the species abbreviation of your input)]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/, This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]
    [-T number of THREADS to use in BLAST search. Default = 8]

[-g runs KOBAS identify]
    KOBAS identify options:
    -f FGFILE foreground file, the output of annotate
    -b BGFILE background file, species abbreviation, see this list for species codes: https://www.kegg.jp/kegg/catalog/org_list.html
    -o OUTPUT file (Default is stdout.)
    [-d DB databases for selection, 1-letter abbreviation separated by &quot;/&quot;: K for KEGG PATHWAY, n for PID, b for BioCarta, R for Reactome, B for BioCyc, p for PANTHER,
           o for OMIM, k for KEGG DISEASE, f for FunDO, g for GAD, N for NHGRI GWAS Catalog and G for Gene Ontology, default K/n/b/R/B/p/o/k/f/g/N/]
    [-m METHOD choose statistical test method: b for binomial test, c for chi-square test, h for hypergeometric test / Fisher&#39;s exact test, and x for frequency list,
           default hypergeometric test / Fisher&#39;s exact test
    [-n FDR choose false discovery rate (FDR) correction method: BH for Benjamini and Hochberg, BY for Benjamini and Yekutieli, QVALUE, and None, default BH
    [-c CUTOFF terms with less than cutoff number of genes are not used for statistical tests, default 5]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/. This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]

[-j runs both KOBAS annotate and identify]
</pre></div>
</div>
</div>
<div class="section" id="id4">
<h3><strong>Running KOBAS with Data</strong><a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There are 3 directories built into this container. These directories should be used to mount data.</p>
<ul class="last simple">
<li>/seq_pep</li>
<li>/sqlite3</li>
<li>/work-dir</li>
</ul>
</div>
<div class="section" id="example-pbs-script-for-annotate-process">
<h4><strong>Example PBS Script for Annotate Process</strong><a class="headerlink" href="#example-pbs-script-for-annotate-process" title="Permalink to this headline">¶</a></h4>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N kobas</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/kobas

singularity pull docker://agbase/kobas:3.0.3_0

singularity run <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3 <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir <span class="se">\</span>
kobas_3.0.3_0.sif <span class="se">\</span>
-a
-i AROS1000.fa <span class="se">\</span>
-s dme <span class="se">\</span>
-t fasta:pro <span class="se">\</span>
-o AROS1000 <span class="se">\</span>
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-command">
<h4><strong>Breakdown of Command</strong><a class="headerlink" href="#breakdown-of-command" title="Permalink to this headline">¶</a></h4>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>kobas_3.0.3_0.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-a:</strong> Tells KOBAS to runt he &#8216;annotate&#8217; process.</p>
<p><strong>-i AROS1000.fa:</strong> input file (peptide FASTA)</p>
<p><strong>-s dme:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o AROS1000:</strong> name of output file</p>
<p>For information on output files see <a class="reference internal" href="#annotateresults">annotateresults</a></p>
</div>
</div>
</div>
<div class="section" id="running-kobas-identify-using-singularity">
<h2><strong>Running KOBAS Identify using Singularity</strong><a class="headerlink" href="#running-kobas-identify-using-singularity" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id5">
<h3><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<p><strong>Example PBS script:</strong></p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N kobas</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/kobas

singularity pull docker://agbase/kobas:3.0.3_0

singularity run <span class="se">\</span>
kobas_3.0.3_0.sif <span class="se">\</span>
-h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>Options:
[-h prints this help statement]

[-a runs KOBAS annotate]
KOBAS annotate options:
    -i INFILE can be FASTA or one-per-lineidentifiers. See -t intype for details.
    -s SPECIES 3 or 4 letter species abbreviation (can be found here: ftp://ftp.cbi.pku.edu.cn/pub/KOBAS_3.0_DOWNLOAD/species_abbr.txt or here: https://www.kegg.jp/kegg/catalog/org_list.html)
    -o OUTPUT file (Default is stdout.)
    -t INTYPE (fasta:pro, fasta:nuc, blastout:xml, blastout:tab, id:ncbigi, id:uniprot, id:ensembl, id:ncbigene), default fasta:pro
    [-l LIST available species, or list available databases for a specific species]
    [-e EVALUE expect threshold for BLAST, default 1e-5]
    [-r RANK rank cutoff for valid hits from BLAST result, default is 5]
    [-C COVERAGE subject coverage cutoff for BLAST, default 0]
    [-z ORTHOLOG whether only use orthologs for cross-species annotation or not, default NO (if only using orthologs, please provide the species abbreviation of your input)]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/, This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]
    [-T number of THREADS to use in BLAST search. Default = 8]

[-g runs KOBAS identify]
    KOBAS identify options:
    -f FGFILE foreground file, the output of annotate
    -b BGFILE background file, species abbreviation, see this list for species codes: https://www.kegg.jp/kegg/catalog/org_list.html
    -o OUTPUT file (Default is stdout.)
    [-d DB databases for selection, 1-letter abbreviation separated by &quot;/&quot;: K for KEGG PATHWAY, n for PID, b for BioCarta, R for Reactome, B for BioCyc, p for PANTHER,
           o for OMIM, k for KEGG DISEASE, f for FunDO, g for GAD, N for NHGRI GWAS Catalog and G for Gene Ontology, default K/n/b/R/B/p/o/k/f/g/N/]
    [-m METHOD choose statistical test method: b for binomial test, c for chi-square test, h for hypergeometric test / Fisher&#39;s exact test, and x for frequency list,
           default hypergeometric test / Fisher&#39;s exact test
    [-n FDR choose false discovery rate (FDR) correction method: BH for Benjamini and Hochberg, BY for Benjamini and Yekutieli, QVALUE, and None, default BH
    [-c CUTOFF terms with less than cutoff number of genes are not used for statistical tests, default 5]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/. This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]

[-j runs both KOBAS annotate and identify]
</pre></div>
</div>
</div>
<div class="section" id="running-kobas-identify-with-data">
<h3><strong>Running KOBAS Identify with Data</strong><a class="headerlink" href="#running-kobas-identify-with-data" title="Permalink to this headline">¶</a></h3>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There are 3 directories built into this container. These directories should be used to mount data.</p>
<ul class="last simple">
<li>/seq_pep</li>
<li>/sqlite3</li>
<li>/work-dir</li>
</ul>
</div>
<div class="section" id="example-pbs-script-for-identify-process">
<h4><strong>Example PBS Script for Identify Process</strong><a class="headerlink" href="#example-pbs-script-for-identify-process" title="Permalink to this headline">¶</a></h4>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N kobas</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/kobas

singularity pull docker://agbase/kobas:3.0.3_0

singularity run <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3 <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir <span class="se">\</span>
kobas_3.0.3_0.sif <span class="se">\</span>
-g
-f AROS1000 <span class="se">\</span>
-b dme <span class="se">\</span>
-o ident_out
</pre></div>
</div>
</div>
<div class="section" id="id6">
<h4><strong>Breakdown of Identify Command</strong><a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h4>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>kobas_3.0.3_0.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-g:</strong> Tells KOBAS to runt he &#8216;identify&#8217; process.</p>
<p><strong>-f AROS1000:</strong> output file from &#8216;annotate&#8217;</p>
<p><strong>-b dme:</strong> background; enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-o ident_out:</strong> name of output file</p>
<p>For information on output see <a class="reference internal" href="#identifyresults">identifyresults</a></p>
</div>
<div class="section" id="example-pbs-script-for-annotate-and-identify-pipeline">
<h4><strong>Example PBS Script for Annotate and Identify Pipeline</strong><a class="headerlink" href="#example-pbs-script-for-annotate-and-identify-pipeline" title="Permalink to this headline">¶</a></h4>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N kobas</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/kobas

singularity pull docker://agbase/kobas:3.0.3_0

singularity run <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3 <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir <span class="se">\</span>
kobas_3.0.3_0.sif <span class="se">\</span>
-j
-i AROS1000.fa <span class="se">\</span>
-s dme <span class="se">\</span>
-t fasta:pro <span class="se">\</span>
-o AROS1000 <span class="se">\</span>
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-annotate-and-idenfity-command">
<h4><strong>Breakdown of Annotate and Idenfity Command</strong><a class="headerlink" href="#breakdown-of-annotate-and-idenfity-command" title="Permalink to this headline">¶</a></h4>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/seq_pep:/seq_pep:</strong> tells docker to mount the &#8216;seq_pep&#8217; directory I downloaded to the host machine to the &#8216;/seq_pep&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/sqlite3:/sqlite3:</strong> mounts &#8216;sqlite3&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>kobas_3.0.3_0.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-j:</strong> Tells KOBAS to runt he &#8216;annotate&#8217; process.</p>
<p><strong>-i AROS1000.fa:</strong> input file (peptide FASTA)</p>
<p><strong>-s dme:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If you don&#8217;t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p class="last">If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o AROS1000:</strong> name of output file</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This pipeline will automatically use the output of &#8216;annotate&#8217; as the -f foreground input for &#8216;identify.
This will also use your species option as the -b background input for &#8216;identify&#8217;.</p>
</div>
<p>For information on outputs see <a class="reference internal" href="#annoident">annoident</a>.</p>
</div>
</div>
</div>
<div class="section" id="understanding-your-results">
<h2><strong>Understanding Your Results</strong><a class="headerlink" href="#understanding-your-results" title="Permalink to this headline">¶</a></h2>
<div class="section" id="annotate">
<span id="annotateresults"></span><h3><strong>Annotate</strong><a class="headerlink" href="#annotate" title="Permalink to this headline">¶</a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><strong>seq_pep folder:</strong> This folder contains the BLAST database files used in your analysis.</li>
<li><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</li>
<li><strong>&lt;species&gt;.tsv:</strong> This is the tab-delimited output from the BLAST search. It is unlikely that you will need to look at this file.</li>
<li><strong>&lt;output_file_name_you_provided&gt;:</strong> KOBAS-annotate generates a text file with the name you provide. It has two sections.</li>
</ul>
<p>The first sections looks like this:</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>##dme       Drosophila melanogaster (fruit fly)
##Method: BLAST     Options: evalue &lt;= 1e-05
##Summary:  87 succeed, 0 fail

#Query      Gene ID|Gene name|Hyperlink
lcl|NW_020311285.1_prot_XP_012256083.1_15   dme:Dmel_CG34349|Unc-13-4B|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG34349
lcl|NW_020311286.1_prot_XP_020708336.1_46   dme:Dmel_CG6963|gish|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<p>The second section follows a dashed line and looks like this:</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>--------------------

////
Query:                      lcl|NW_020311285.1_prot_XP_012256083.1_15
Gene:                       dme:Dmel_CG34349        Unc-13-4B
Entrez Gene ID:             43002
////
Query:                      lcl|NW_020311286.1_prot_XP_020708336.1_46
Gene:                       dme:Dmel_CG6963 gish
Entrez Gene ID:             49701
Pathway:                    Hedgehog signaling pathway - fly        KEGG PATHWAY    dme04341
////
Query:                      lcl|NW_020311285.1_prot_XP_020707987.1_39
Gene:                       dme:Dmel_CG30403
Entrez Gene ID:             246595
////
Query:                      lcl|NW_020311285.1_prot_XP_020707989.1_40
Gene:                       dme:Dmel_CG6148 Past1
Entrez Gene ID:             41569
Pathway:                    Endocytosis     KEGG PATHWAY    dme04144
                            Hemostasis      Reactome        R-DME-109582
                            Factors involved in megakaryocyte development and platelet production   Reactome        R-DME-98323
</pre></div>
</div>
</div>
<div class="section" id="identifyresults">
<span id="identify"></span><h3><strong>Identify</strong><a class="headerlink" href="#identifyresults" title="Permalink to this headline">¶</a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</li>
<li><strong>&lt;output_file_name_you_provided&gt;:</strong> KOBAS identify generates a text file with the name you provide.</li>
</ul>
<div class="highlight-none"><div class="highlight"><pre><span></span>##Databases: PANTHER, KEGG PATHWAY, Reactome, BioCyc
##Statistical test method: hypergeometric test / Fisher&#39;s exact test
##FDR correction method: Benjamini and Hochberg

#Term       Database        ID      Input number    Background number       P-Value Corrected P-Value       Input   Hyperlink
Hedgehog signaling pathway - fly    KEGG PATHWAY    dme04341        12      33      3.20002656734e-18       1.76001461204e-16       lcl|NW_020311286.1_prot_XP_012256678.1_51|lcl|NW_020311286.1_prot_XP_025602973.1_48|lcl|NW_020311286.1_prot_XP_012256683.1_52|lcl|NW_020311286.1_prot_XP_012256679.1_55|lcl|NW_020311286.1_prot_XP_012256674.1_54|lcl|NW_020311286.1_prot_XP_020708336.1_46|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311286.1_prot_XP_012256682.1_53|lcl|NW_020311286.1_prot_XP_025603025.1_47|lcl|NW_020311286.1_prot_XP_020708334.1_49|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_020708333.1_50 http://www.genome.jp/kegg-bin/show_pathway?dme04341/dme:Dmel_CG6963%09red/dme:Dmel_CG6054%09red
Hedgehog signaling pathway  PANTHER P00025  6       13      3.6166668094e-10        9.94583372585e-09       lcl|NW_020311286.1_prot_XP_025602279.1_78|lcl|NW_020311286.1_prot_XP_025602289.1_76|lcl|NW_020311286.1_prot_XP_025602264.1_79|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_012256943.1_77     http://www.pantherdb.org/pathway/pathwayDiagram.jsp?catAccession=P00025
Signaling by NOTCH2 Reactome        R-DME-1980145   3       8       2.00259649553e-05       0.000275357018136       lcl|NW_020311285.1_prot_XP_012256118.1_28|lcl|NW_020311285.1_prot_XP_012256117.1_27|lcl|NW_020311285.1_prot_XP_012256119.1_26   http://www.reactome.org/cgi-bin/eventbrowser_st_id?ST_ID=R-DME-1980145
</pre></div>
</div>
<p id="annoident"><strong>Annotate and Identify Pipeline</strong></p>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><strong>logs folder:</strong> This folder contains the &#8216;conder_stderr&#8217; and &#8216;condor_stdout&#8217; files. The files record feedback, progress and, importantly, any errors the app encountered during the analysis. You won&#8217;t normally need to look at these but they are very helpful in figuring out what may have happened if your output doesn&#8217;t look like you expected.</li>
<li><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</li>
<li><strong>seq_pep folder:</strong> This folder contains the BLAST database files used in your analysis.</li>
<li><strong>&lt;species&gt;.tsv:</strong> This is the tab-delimited output from the BLAST search. It is unlikely that you will need to look at this file.</li>
<li><strong>&lt;basename&gt;_annotate_out.txt:</strong> KOBAS annotate generates a text file with the name you provide. It has two sections.</li>
</ul>
<p>The first sections looks like this:</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>##dme       Drosophila melanogaster (fruit fly)
##Method: BLAST     Options: evalue &lt;= 1e-05
##Summary:  87 succeed, 0 fail

#Query      Gene ID|Gene name|Hyperlink
lcl|NW_020311285.1_prot_XP_012256083.1_15   dme:Dmel_CG34349|Unc-13-4B|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG34349
lcl|NW_020311286.1_prot_XP_020708336.1_46   dme:Dmel_CG6963|gish|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<p>The second section follows a dashed line and looks like this:</p>
<div class="highlight-none"><div class="highlight"><pre><span></span>--------------------

////
Query:                      lcl|NW_020311285.1_prot_XP_012256083.1_15
Gene:                       dme:Dmel_CG34349        Unc-13-4B
Entrez Gene ID:             43002
////
Query:                      lcl|NW_020311286.1_prot_XP_020708336.1_46
Gene:                       dme:Dmel_CG6963 gish
Entrez Gene ID:             49701
Pathway:                    Hedgehog signaling pathway - fly        KEGG PATHWAY    dme04341
////
Query:                      lcl|NW_020311285.1_prot_XP_020707987.1_39
Gene:                       dme:Dmel_CG30403
Entrez Gene ID:             246595
////
Query:                      lcl|NW_020311285.1_prot_XP_020707989.1_40
Gene:                       dme:Dmel_CG6148 Past1
Entrez Gene ID:             41569
Pathway:                    Endocytosis     KEGG PATHWAY    dme04144
                            Hemostasis      Reactome        R-DME-109582
                            Factors involved in megakaryocyte development and platelet production   Reactome        R-DME-98323mel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<ul class="simple">
<li><strong>&lt;basename&gt;_identify_out.txt:</strong> KOBAS identify generates a text file with the name you provide.</li>
</ul>
<div class="highlight-none"><div class="highlight"><pre><span></span>##Databases: PANTHER, KEGG PATHWAY, Reactome, BioCyc
##Statistical test method: hypergeometric test / Fisher&#39;s exact test
##FDR correction method: Benjamini and Hochberg

#Term       Database        ID      Input number    Background number       P-Value Corrected P-Value       Input   Hyperlink
Hedgehog signaling pathway - fly    KEGG PATHWAY    dme04341        12      33      3.20002656734e-18       1.76001461204e-16       lcl|NW_020311286.1_prot_XP_012256678.1_51|lcl|NW_020311286.1_prot_XP_025602973.1_48|lcl|NW_020311286.1_prot_XP_012256683.1_52|lcl|NW_020311286.1_prot_XP_012256679.1_55|lcl|NW_020311286.1_prot_XP_012256674.1_54|lcl|NW_020311286.1_prot_XP_020708336.1_46|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311286.1_prot_XP_012256682.1_53|lcl|NW_020311286.1_prot_XP_025603025.1_47|lcl|NW_020311286.1_prot_XP_020708334.1_49|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_020708333.1_50 http://www.genome.jp/kegg-bin/show_pathway?dme04341/dme:Dmel_CG6963%09red/dme:Dmel_CG6054%09red
Hedgehog signaling pathway  PANTHER P00025  6       13      3.6166668094e-10        9.94583372585e-09       lcl|NW_020311286.1_prot_XP_025602279.1_78|lcl|NW_020311286.1_prot_XP_025602289.1_76|lcl|NW_020311286.1_prot_XP_025602264.1_79|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_012256943.1_77     http://www.pantherdb.org/pathway/pathwayDiagram.jsp?catAccession=P00025
Signaling by NOTCH2 Reactome        R-DME-1980145   3       8       2.00259649553e-05       0.000275357018136       lcl|NW_020311285.1_prot_XP_012256118.1_28|lcl|NW_020311285.1_prot_XP_012256117.1_27|lcl|NW_020311285.1_prot_XP_012256119.1_26   http://www.reactome.org/cgi-bin/eventbrowser_st_id?ST_ID=R-DME-1980145
</pre></div>
</div>
<p><a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="using_kobas_in_cyverse_DE.html" class="btn btn-neutral" title="KOBAS on CyVerse" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Fiona M. McCarthy.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>