<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>KOBAS on the Command Line &mdash; AgBase 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="KOBAS on CyVerse" href="using_kobas_in_cyverse_DE.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            AgBase
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Agbase Home</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../agbase/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">InterProScan</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_in_cyverse_DE.html"><strong>InterProScan on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Combine GAFs</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cyverse.html"><strong>Combine GAFs on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cmd.html"><strong>Combine GAFs on the Command Line</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">KOBAS</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>KOBAS on the Command Line</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-databases"><strong>Getting the databases</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#container-technologies"><strong>Container Technologies</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-kobas-using-docker"><strong>Running KOBAS using Docker</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-kobas-container"><strong>Getting the KOBAS container</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#getting-the-help-and-usage-statement"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#annotate-example-command"><strong>Annotate Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#identify-example-command"><strong>Identify Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#annotate-and-identify-pipeline-example-command"><strong>Annotate and Identify Pipeline Example Command</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#running-kobas-using-singularity"><strong>Running KOBAS using Singularity</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id3"><strong>Getting the KOBAS container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-kobas-with-data"><strong>Running KOBAS with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-slurm-script-for-annotate-process"><strong>Example Slurm Script for Annotate Process</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#example-slurm-script-for-identify-process"><strong>Example Slurm Script for Identify Process</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#example-slurm-script-for-annotate-and-identify-pipeline"><strong>Example Slurm Script for Annotate and Identify Pipeline</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#understanding-your-results"><strong>Understanding Your Results</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#annotate"><strong>Annotate</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#identifyresults"><strong>Identify</strong></a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">AgBase</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active"><strong>KOBAS on the Command Line</strong></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/kobas/using_kobas_cmd.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="kobas-on-the-command-line">
<h1><strong>KOBAS on the Command Line</strong><a class="headerlink" href="#kobas-on-the-command-line" title="Permalink to this headline"></a></h1>
<section id="getting-the-databases">
<h2><strong>Getting the databases</strong><a class="headerlink" href="#getting-the-databases" title="Permalink to this headline"></a></h2>
<p><a class="reference external" href="file:///home/amcooksey/Documents/USDA_i5K/Docs/_build/kobas/intro.html#getting-the-kobas-databases">No longer required as of version 3.0.3_3</a>.</p>
</section>
<section id="container-technologies">
<h2><strong>Container Technologies</strong><a class="headerlink" href="#container-technologies" title="Permalink to this headline"></a></h2>
<p>KOBAS is provided as a Docker container.</p>
<p>A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another.</p>
<p>There are two major containerization technologies: <strong>Docker</strong> and <strong>Apptainer (Singularity)</strong>.</p>
<p>Docker containers can be run with either technology.</p>
</section>
<section id="running-kobas-using-docker">
<h2><strong>Running KOBAS using Docker</strong><a class="headerlink" href="#running-kobas-using-docker" title="Permalink to this headline"></a></h2>
<div class="admonition-about-docker admonition">
<p class="admonition-title">About Docker</p>
<ul class="simple">
<li><p>Docker must be installed on the computer you wish to use for your analysis.</p></li>
<li><p>To run Docker you must have ‘root’ (admin) permissions (or use sudo).</p></li>
<li><p>Docker will run all containers as ‘root’. This makes Docker incompatible with HPC systems (see Singularity below).</p></li>
<li><p>Docker can be run on your local computer, a server, a cloud virtual machine etc.</p></li>
<li><p>For more information on installing Docker on other systems:  <a class="reference external" href="https://docs.docker.com/engine/install/">Installing Docker</a>.</p></li>
</ul>
</div>
<section id="getting-the-kobas-container">
<h3><strong>Getting the KOBAS container</strong><a class="headerlink" href="#getting-the-kobas-container" title="Permalink to this headline"></a></h3>
<p>The KOBAS tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/kobas">KOBAS container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>docker pull agbase/kobas:3.0.3_3
</pre></div>
</div>
<div class="admonition-remember admonition">
<p class="admonition-title">Remember</p>
<p>You must have root permissions or use sudo, like so:</p>
<p>sudo docker pull agbase/kobas:3.0.3_3</p>
</div>
<section id="getting-the-help-and-usage-statement">
<h4><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#getting-the-help-and-usage-statement" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run --rm agbase/kobas:3.0.3_3 -h
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>The /work-dir directory is built into this container and should be used to mount your data.</p>
</div>
<p>KOBAS can perform two tasks
- annotate (-a)
- identify (enrichment) (-g)</p>
<p>KOBAS can also run both tasks with a single command (-j).</p>
</section>
<section id="annotate-example-command">
<h4><strong>Annotate Example Command</strong><a class="headerlink" href="#annotate-example-command" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run <span class="se">\</span>
--rm <span class="se">\</span>
-v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir <span class="se">\</span>
agbase/kobas:3.0.3_3 <span class="se">\</span>
-a <span class="se">\</span>
-i GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-s sce <span class="se">\</span>
-t fasta:pro <span class="se">\</span>
-o GCF_001298625.1
</pre></div>
</div>
<section id="command-explained">
<h5><strong>Command Explained</strong><a class="headerlink" href="#command-explained" title="Permalink to this headline"></a></h5>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>–rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ inside the container</p>
<p><strong>agbase/kobas:3.0.3_3:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-a:</strong> Tells KOBAS to run the ‘annotate’ process.</p>
<p><strong>-i GCF_001298625.1_SEUB3.0_protein.faa:</strong> input file (protein FASTA).</p>
<p><strong>-s sce:</strong> Enter the species code for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o GCF_001298625.1:</strong> prefix for the output file names</p>
<p>Reference <a class="reference internal" href="#understanding-results">Understanding results</a>.</p>
</section>
</section>
<section id="identify-example-command">
<h4><strong>Identify Example Command</strong><a class="headerlink" href="#identify-example-command" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run <span class="se">\</span>
--rm <span class="se">\</span>
-v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir <span class="se">\</span>
agbase/kobas:3.0.3_3 <span class="se">\</span>
-g <span class="se">\</span>
-f GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-b sce <span class="se">\</span>
-o ident_out
</pre></div>
</div>
<section id="id1">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id1" title="Permalink to this headline"></a></h5>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>–rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ in the container</p>
<p><strong>agbase/kobas:3.0.3_3:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-g:</strong> Tells KOBAS to runt he ‘identify’ process.</p>
<p><strong>-f GCF_001298625.1_SEUB3.0_protein.faa:</strong> output file from KOBAS annotate</p>
<p><strong>-b sce:</strong> background; enter the species code for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-o ident_out:</strong> basename of output file</p>
<p>Reference <a class="reference internal" href="#understanding-results">Understanding results</a>.</p>
</section>
</section>
<section id="annotate-and-identify-pipeline-example-command">
<h4><strong>Annotate and Identify Pipeline Example Command</strong><a class="headerlink" href="#annotate-and-identify-pipeline-example-command" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo docker run <span class="se">\</span>
--rm <span class="se">\</span>
-v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir <span class="se">\</span>
agbase/kobas:3.0.3_3 <span class="se">\</span>
-j <span class="se">\</span>
-i GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-s sce <span class="se">\</span>
-t fasta:pro
-o GCF_001298625.1
</pre></div>
</div>
<section id="id2">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id2" title="Permalink to this headline"></a></h5>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>–rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ in the container</p>
<p><strong>agbase/kobas:3.0.3_3:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-j:</strong> Tells KOBAS to run both the ‘annotate’ and ‘identify’ processes.</p>
<p><strong>-i GCF_001298625.1_SEUB3.0_protein.faa:</strong> input file (protein FASTA)</p>
<p><strong>-s sce:</strong> Enter the species code for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o GCF_001298625.1:</strong> basename of output files</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This pipeline will automatically use the output of ‘annotate’ as the -f foreground input for ‘identify’.
This will also use your species option as the -b background input for ‘identify’.</p>
</div>
<p>Reference <a class="reference internal" href="#understanding-results">Understanding results</a>.</p>
</section>
</section>
</section>
</section>
<section id="running-kobas-using-singularity">
<h2><strong>Running KOBAS using Singularity</strong><a class="headerlink" href="#running-kobas-using-singularity" title="Permalink to this headline"></a></h2>
<div class="admonition-about-singularity-now-apptainer admonition">
<p class="admonition-title">About Singularity (now Apptainer)</p>
<ul class="simple">
<li><p>does not require ‘root’ permissions</p></li>
<li><p>runs all containers as the user that is logged into the host machine</p></li>
<li><p>HPC systems are likely to have Singularity installed and are unlikely to object if asked to install it (no guarantees).</p></li>
<li><p>can be run on any machine where it is installed</p></li>
<li><p>more information about <a class="reference external" href="https://apptainer.org/docs-legacy">installing Singularity</a></p></li>
<li><p>This tool was tested using Singularity 3.10.2.</p></li>
</ul>
</div>
<div class="admonition-hpc-job-schedulers admonition">
<p class="admonition-title">HPC Job Schedulers</p>
<p>Although Singularity can be installed on any computer this documentation assumes it will be run on an HPC system. The tool was tested on a Slurm system and the job submission scripts below reflect that. Submission scripts will need to be modified for use with other job scheduler systems.</p>
</div>
<section id="id3">
<h3><strong>Getting the KOBAS container</strong><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h3>
<p>The KOBAS tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/kobas">KOBAS container</a></p>
<p><strong>Example Slurm script:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=kobas</span>
<span class="c1">#SBATCH --ntasks=8</span>
<span class="c1">#SBATCH --time=2:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>

module load singularity

<span class="nb">cd</span> /location/where/your/want/to/save/file

singularity pull docker://agbase/kobas:3.0.3_3
</pre></div>
</div>
</section>
<section id="running-kobas-with-data">
<h3><strong>Running KOBAS with Data</strong><a class="headerlink" href="#running-kobas-with-data" title="Permalink to this headline"></a></h3>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>There /work-dir directory is built into this container and should be used to mount data.</p>
</div>
<section id="example-slurm-script-for-annotate-process">
<h4><strong>Example Slurm Script for Annotate Process</strong><a class="headerlink" href="#example-slurm-script-for-annotate-process" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=kobas</span>
<span class="c1">#SBATCH --ntasks=8</span>
<span class="c1">#SBATCH --time=2:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>

module load singularity

<span class="nb">cd</span> /directory/you/want/to/work/in

singularity run <span class="se">\</span>
-B /directory/you/want/to/work/in:/work-dir <span class="se">\</span>
/path/to/your/copy/kobas_3.0.3_3.sif <span class="se">\</span>
-a <span class="se">\</span>
-i GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-s sce <span class="se">\</span>
-t fasta:pro <span class="se">\</span>
-o GCF_001298625.1
</pre></div>
</div>
<section id="id5">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id5" title="Permalink to this headline"></a></h5>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /project/nal_genomics/amanda.cooksey/protein_sets/saceub/KOBAS:/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ in the container</p>
<p><strong>/path/to/your/copy/kobas_3.0.3_3.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-a:</strong> Tells KOBAS to run the ‘annotate’ process.</p>
<p><strong>-i GCF_001298625.1_SEUB3.0_protein.faa:</strong> input file (protein FASTA)</p>
<p><strong>-s sce:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o GCF_001298625.1:</strong> name of output file</p>
<p>Reference <a class="reference internal" href="#understanding-results">Understanding results</a>.</p>
</section>
</section>
<section id="example-slurm-script-for-identify-process">
<h4><strong>Example Slurm Script for Identify Process</strong><a class="headerlink" href="#example-slurm-script-for-identify-process" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=kobas</span>
<span class="c1">#SBATCH --ntasks=8</span>
<span class="c1">#SBATCH --time=2:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>

module load singularity

<span class="nb">cd</span> /location/where/your/want/to/save/file

singularity pull docker://agbase/kobas:3.0.3_3

singularity run <span class="se">\</span>
-B /directory/you/want/to/work/in:/work-dir <span class="se">\</span>
kobas_3.0.3_3.sif <span class="se">\</span>
-g <span class="se">\</span>
-f GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-b sce <span class="se">\</span>
-o ident_out
</pre></div>
</div>
<section id="id6">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id6" title="Permalink to this headline"></a></h5>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /project/nal_genomics/amanda.cooksey/protein_sets/saceub/KOBAS:/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ in the container</p>
<p><strong>kobas_3.0.3_3.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-g:</strong> Tells KOBAS to run the ‘identify’ process.</p>
<p><strong>-f GCF_001298625.1_SEUB3.0_protein.faa:</strong> output file from ‘annotate’</p>
<p><strong>-b sce:</strong> background; enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-o ident_out:</strong> name of output file</p>
<p>Reference <a class="reference internal" href="#understanding-results">Understanding results</a>.</p>
</section>
</section>
<section id="example-slurm-script-for-annotate-and-identify-pipeline">
<h4><strong>Example Slurm Script for Annotate and Identify Pipeline</strong><a class="headerlink" href="#example-slurm-script-for-annotate-and-identify-pipeline" title="Permalink to this headline"></a></h4>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH --job-name=kobas</span>
<span class="c1">#SBATCH --ntasks=8</span>
<span class="c1">#SBATCH --time=2:00:00</span>
<span class="c1">#SBATCH --partition=short</span>
<span class="c1">#SBATCH --account=nal_genomics</span>

module load singularity

<span class="nb">cd</span> /location/where/your/want/to/save/file

singularity pull docker://agbase/kobas:3.0.3_3

singularity run <span class="se">\</span>
-B /directory/you/want/to/work/in:/work-dir <span class="se">\</span>
kobas_3.0.3_3.sif <span class="se">\</span>
-j <span class="se">\</span>
-i GCF_001298625.1_SEUB3.0_protein.faa <span class="se">\</span>
-s sce <span class="se">\</span>
-t fasta:pro <span class="se">\</span>
-o GCF_001298625.1
</pre></div>
</div>
<section id="id7">
<h5><strong>Command Explained</strong><a class="headerlink" href="#id7" title="Permalink to this headline"></a></h5>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/kobas:/work-dir:</strong> mounts my current working directory on the host machine to ‘/work-dir’ in the container</p>
<p><strong>kobas_3.0.3_3.sif:</strong> the name of the Singularity image to use</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>All the options supplied after the image name are KOBAS options</p>
</div>
<p><strong>-j:</strong> Tells KOBAS to runt he ‘annotate’ process.</p>
<p><strong>-i GCF_001298625.1_SEUB3.0_protein.faa:</strong> input file (protein FASTA)</p>
<p><strong>-s sce:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>-t:</strong> input file type; in this case, protein FASTA.</p>
<p><strong>-o GCF_001298625.1:</strong> name of output file</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This pipeline will automatically use the output of ‘annotate’ as the -f foreground input for ‘identify’.
This will also use your species option as the -b background input for ‘identify’.</p>
</div>
</section>
</section>
</section>
</section>
<section id="understanding-your-results">
<span id="understanding-results"></span><h2><strong>Understanding Your Results</strong><a class="headerlink" href="#understanding-your-results" title="Permalink to this headline"></a></h2>
<section id="annotate">
<h3><strong>Annotate</strong><a class="headerlink" href="#annotate" title="Permalink to this headline"></a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><p><strong>&lt;species&gt;.tsv:</strong> This is the tab-separated output from the BLAST search. It is unlikely that you will need to look at this file.</p></li>
<li><p><strong>&lt;basename&gt;:</strong> KOBAS-annotate generates a text file with the name you provide. It has two sections (detailed below).</p></li>
<li><p><strong>&lt;basename&gt;_KOBAS_acc_pathways.tsv:</strong> Our post-processing script creates this tab-separated file. It lists each accession from your data and all of the pathways to which they were annotated.</p></li>
<li><p><strong>&lt;basename&gt;_KOBAS_pathways_acc.tsv:</strong> Our post-processing script creates this tab-separated file. It lists each pathway annotated to your data with all of the accessions annotated to that pathway.</p></li>
</ul>
<p>The &lt;basename&gt; file has two sections.
The first section looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#Query      Gene ID|Gene name|Hyperlink</span>
XP_018220118.1      sce:YMR059W<span class="p">|</span>SEN15<span class="p">|</span>http://www.genome.jp/dbget-bin/www_bget?sce:YMR059W
XP_018221352.1      sce:YJR050W<span class="p">|</span>ISY1, NTC30, UTR3<span class="p">|</span>http://www.genome.jp/dbget-bin/www_bget?sce:YJR050W
XP_018224031.1      sce:YDR513W<span class="p">|</span>GRX2, TTR1<span class="p">|</span>http://www.genome.jp/dbget-bin/www_bget?sce:YDR513W
XP_018222559.1      sce:YFR024C-A<span class="p">|</span>LSB3, YFR024C<span class="p">|</span>http://www.genome.jp/dbget-bin/www_bget?sce:YFR024C-A
XP_018221254.1      sce:YJL070C<span class="o">||</span>http://www.genome.jp/dbget-bin/www_bget?sce:YJL070C
</pre></div>
</div>
<p>The second section follows a dashed line and looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>////
Query:                  XP_018222878.1
Gene:                   sce:YDL220C     CDC13, EST4
Entrez Gene ID:         <span class="m">851306</span>
////
Query:                  XP_018219412.1
Gene:                   sce:YOR204W     DED1, SPP81
Entrez Gene ID:         <span class="m">854379</span>
Pathway:                Innate Immune System    Reactome        R-SCE-168249
                        Immune System   Reactome        R-SCE-168256
                        Neutrophil degranulation        Reactome        R-SCE-6798695
</pre></div>
</div>
<p>&lt;basename&gt;_KOBAS_acc_pathways.tsv looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>XP_018220118.1      BioCyc:PWY-6689
XP_018221352.1      Reactome:R-SCE-6782135,KEGG:sce03040,Reactome:R-SCE-73894,Reactome:R-SCE-5696398,Reactome:R-SCE-6782210,Reactome:R-SCE-6781827
XP_018224031.1      BioCyc:GLUT-REDOX-PWY,BioCyc:PWY3O-592
</pre></div>
</div>
<p>&lt;basename&gt;_KOBAS_pathways_acc.tsv looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>BioCyc:PWY3O-0  XP_018222002.1,XP_018222589.1
KEGG:sce00440   XP_018222406.1,XP_018219751.1,XP_018222229.1
Reactome:R-SCE-416476   XP_018223583.1,XP_018221814.1,XP_018222685.1,XP_018220832.1,XP_018219073.1,XP_018218776.1,XP_018223466.1,XP_018223545.1,XP_018222256.1
Reactome:R-SCE-418346   XP_018220070.1,XP_018221774.1,XP_018221826.1,XP_018220071.1,XP_018222218.1,XP_018220541.1,XP_018219550.1
</pre></div>
</div>
</section>
<section id="identifyresults">
<span id="identify"></span><h3><strong>Identify</strong><a class="headerlink" href="#identifyresults" title="Permalink to this headline"></a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><p><strong>&lt;output_file_name_you_provided&gt;:</strong> KOBAS identify generates a text file with the name you provide.</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">##Databases: PANTHER, KEGG PATHWAY, Reactome, BioCyc</span>
<span class="c1">##Statistical test method: hypergeometric test / Fisher&#39;s exact test</span>
<span class="c1">##FDR correction method: Benjamini and Hochberg</span>

<span class="c1">#Term   Database        ID      Input number    Background number       P-Value Corrected P-Value       Input   Hyperlink</span>
Metabolic pathways      KEGG PATHWAY    sce01100        <span class="m">714</span>     <span class="m">754</span>     <span class="m">0</span>.00303590229485        <span class="m">0</span>.575578081959  XP_018221856.1<span class="p">|</span>XP_018220917.1<span class="p">|</span>XP_018222719.1<span class="p">|</span>...link
Metabolism      Reactome        R-SCE-1430728   <span class="m">419</span>     <span class="m">438</span>     <span class="m">0</span>.0147488189928 <span class="m">0</span>.575578081959  XP_018221856.1<span class="p">|</span>XP_018221742.1<span class="p">|</span>XP_018219354.1<span class="p">|</span>XP_018221740.1<span class="p">|</span>...link
Immune System   Reactome        R-SCE-168256    <span class="m">304</span>     <span class="m">315</span>     <span class="m">0</span>.0267150787723 <span class="m">0</span>.575578081959  XP_018223955.1<span class="p">|</span>XP_018222962.1<span class="p">|</span>XP_018223268.1<span class="p">|</span>XP_018222956.1<span class="p">|</span>...link
</pre></div>
</div>
<p><a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="using_kobas_in_cyverse_DE.html" class="btn btn-neutral float-left" title="KOBAS on CyVerse" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Fiona M. McCarthy.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>