<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>KOBAS on CyVerse &mdash; AgBase 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="KOBAS on the Command Line" href="using_kobas_cmd.html" />
    <link rel="prev" title="Intro" href="intro.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            AgBase
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Agbase Home</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../agbase/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_ceres.html"><strong>GOanna on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">InterProScan</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_in_cyverse_DE.html"><strong>InterProScan on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_ceres.html"><strong>InterProScan on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Combine GAFs</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cyverse.html"><strong>Combine GAFs on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cmd.html"><strong>Combine GAFs on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_ceres.html"><strong>Combine GAFs on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">KOBAS</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>KOBAS on CyVerse</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#accessing-kobas-in-the-discovery-environment"><strong>Accessing KOBAS in the Discovery Environment</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1"><strong>KOBAS annotate 3.0.3</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#launching-the-app"><strong>Launching the app</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#input"><strong>Input</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#options"><strong>Options</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#output"><strong>Output</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#understanding-your-annotate-results"><strong>Understanding Your Annotate Results</strong></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id2"><strong>KOBAS identify 3.0.3</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id3"><strong>Launching the App</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id4"><strong>Input</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id5"><strong>Options</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id6"><strong>Output</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#understanding-your-identify-results"><strong>Understanding Your Identify Results</strong></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#id7"><strong>KOBAS annotate and identify 3.0.3</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id8"><strong>Launching the App</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id9"><strong>Input</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#annotate-options"><strong>Annotate Options</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#identify-options"><strong>Identify Options</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id10"><strong>Output</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#understanding-your-annotate-and-identify-pipeline-results"><strong>Understanding Your Annotate and Identify Pipeline Results</strong></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="using_kobas_cmd.html"><strong>KOBAS on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_kobas_ceres.html"><strong>KOBAS on the ARS Ceres HPC</strong></a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">AgBase</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active"><strong>KOBAS on CyVerse</strong></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/kobas/using_kobas_in_cyverse_DE.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="kobas-on-cyverse">
<h1><strong>KOBAS on CyVerse</strong><a class="headerlink" href="#kobas-on-cyverse" title="Permalink to this headline"></a></h1>
<section id="accessing-kobas-in-the-discovery-environment">
<h2><strong>Accessing KOBAS in the Discovery Environment</strong><a class="headerlink" href="#accessing-kobas-in-the-discovery-environment" title="Permalink to this headline"></a></h2>
<ol class="arabic simple">
<li><p><a class="reference external" href="user.cyverse.org">Create an account on CyVerse</a> (free)</p></li>
<li><p>Open the CyVerse Discovery Environment (DE) and login with your CyVerse credentials.</p></li>
<li><p>If you are new to the Discovery Environment (DE) the user guide can be found <a class="reference external" href="https://learning.cyverse.org/projects/discovery-environment-guide/en/latest/">here</a>.</p></li>
<li><p>Click on the ‘Data’ button at the left side of the screen to access your files/folders. Upload your data to the DE.</p></li>
<li><p>To access the KOBAS apps click on the ‘Apps’ button at the left side of the DE.</p></li>
<li><p>Search for ‘kobas” in the search bar at the top of the ‘apps’ window (see below). The contents of the folder will appear in the main pane of the window; click on the name to open the app.</p></li>
</ol>
<p>The KOBAS apps are called:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=070f519e-983f-11e9-b659-008cfa5ae621&amp;system-id=de">KOBAS annotate 3.0.3</a></p></li>
<li><p><a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=9e0a429c-dee0-11e9-948a-008cfa5ae621&amp;system-id=de">KOBAS identify 3.0.3</a></p></li>
<li><p><a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=2959dcb4-d0d0-11e9-9f25-008cfa5ae621&amp;system-id=de">KOBAS annotate and identify 3.0.3</a></p></li>
</ul>
<p><a class="reference internal" href="../_images/find_kobas.png"><img alt="find_kobas" src="../_images/find_kobas.png" style="width: 700px;" /></a></p>
<div class="admonition-find-apps-easily-with-communities admonition">
<p class="admonition-title">Find Apps Easily with ‘Communities’</p>
<p>The KOBAS apps belong to the ‘i5k’ and ‘AgBase’ communties. You can join either of these communities and they will appear in the left-hand pane of your ‘Apps’ window (see above).
To join a community:</p>
<ol class="arabic simple">
<li><p>Click on the person icon in the top-right corner of the Discovery Environment window</p></li>
<li><p>Select ‘Communities’</p></li>
<li><p>In the ‘Communities’ window choose ‘all communities’ from the drop-down list. A list of communities will appear in the main pane of this window.</p></li>
<li><p>Select the one you wish to join by clicking on it and then clicking on the ‘join’ button.</p></li>
</ol>
</div>
</section>
<section id="id1">
<h2><strong>KOBAS annotate 3.0.3</strong><a class="headerlink" href="#id1" title="Permalink to this headline"></a></h2>
<section id="launching-the-app">
<h3><strong>Launching the app</strong><a class="headerlink" href="#launching-the-app" title="Permalink to this headline"></a></h3>
<p><a class="reference internal" href="../_images/kobas_anno.png"><img alt="kobas_anno" src="../_images/kobas_anno.png" style="width: 700px;" /></a></p>
<p><strong>Analysis Name: KOBAS_annotate_3.0.3_analysis_1:</strong>
This menu is used to name the job you will run so that you can find it later.
Analysis Name: The default name is “KOBAS_annotate_3.0.3_analysis1”. We recommend changing the ‘analysis1’ portion of this to reflect the data you are running.</p>
<p><strong>Comments:</strong>
(Optional) You can add additional information in the comments section to distinguish your analyses further.</p>
<p><strong>Select output folder:</strong>
This is where your results will be placed. The default (recommended) is your ‘analyses’ folder.</p>
<p><strong>Retain Inputs:</strong>
Enabling this flag will copy all the input files into the analysis result folder.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Selecting this option will rapidly consume your allocated space. It is not recommended. Your inputs will always remain available in the folder in which you stored them.</p>
</div>
<section id="input">
<h4><strong>Input</strong><a class="headerlink" href="#input" title="Permalink to this headline"></a></h4>
<p><strong>Input File:</strong> Use the ‘browse’ button on the right side of the field to navigate to your input file.</p>
<p><strong>Input File Type:</strong> Select your input file type from the drop-down list. If your file type isn’t there then the app does not support that file type.</p>
</section>
<section id="options">
<h4><strong>Options</strong><a class="headerlink" href="#options" title="Permalink to this headline"></a></h4>
<p><strong>Species Code:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a>
<strong>Not all KEGG species are available through KOBAS.</strong></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>E value:</strong> This is the evalue to use in the BLAST search. The default is 1e-5.</p>
<p><strong>Rank:</strong> rank cutoff for valid hits from BLAST result. Default is 5.</p>
<p><strong>Covergage:</strong> subject coverage cutoff for BLAST. Default is 0.</p>
<p><strong>Ortholog:</strong> when checked KOBAS will only use orthologs for cross species annotation.</p>
</section>
<section id="output">
<h4><strong>Output</strong><a class="headerlink" href="#output" title="Permalink to this headline"></a></h4>
<p><strong>Output File Name:</strong> Provide an output file name .</p>
<p>For information on outputs see <a class="reference internal" href="using_kobas_cmd.html#annotateresults"><span class="std std-ref">Understanding Your Results: Annotate</span></a></p>
</section>
</section>
<section id="understanding-your-annotate-results">
<h3><strong>Understanding Your Annotate Results</strong><a class="headerlink" href="#understanding-your-annotate-results" title="Permalink to this headline"></a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><p><strong>logs folder:</strong> This folder contains the ‘conder_stderr’ and ‘condor_stdout’ files. The files record feedback, progress and, importantly, any errors the app encountered during the analysis. You won’t normally need to look at these but they are very helpful in figuring out what may have happened if your output doesn’t look like you expected.</p></li>
<li><p><strong>seq_pep folder:</strong> This folder contains the BLAST database files used in your analysis.</p></li>
<li><p><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</p></li>
<li><p><strong>&lt;species&gt;.tsv:</strong> This is the tab-delimited output from the BLAST search. It is unlikely that you will need to look at this file.</p></li>
<li><p><strong>&lt;output_file_name_you_provided&gt;:</strong> KOBAS-annotate generates a text file with the name you provide. It has two sections.</p></li>
</ul>
<p>The first sections looks like this:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>##dme       Drosophila melanogaster (fruit fly)
##Method: BLAST     Options: evalue &lt;= 1e-05
##Summary:  87 succeed, 0 fail

#Query      Gene ID|Gene name|Hyperlink
lcl|NW_020311285.1_prot_XP_012256083.1_15   dme:Dmel_CG34349|Unc-13-4B|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG34349
lcl|NW_020311286.1_prot_XP_020708336.1_46   dme:Dmel_CG6963|gish|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<p>The second section follows a dashed line and looks like this:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>--------------------

////
Query:                      lcl|NW_020311285.1_prot_XP_012256083.1_15
Gene:                       dme:Dmel_CG34349        Unc-13-4B
Entrez Gene ID:             43002
////
Query:                      lcl|NW_020311286.1_prot_XP_020708336.1_46
Gene:                       dme:Dmel_CG6963 gish
Entrez Gene ID:             49701
Pathway:                    Hedgehog signaling pathway - fly        KEGG PATHWAY    dme04341
////
Query:                      lcl|NW_020311285.1_prot_XP_020707987.1_39
Gene:                       dme:Dmel_CG30403
Entrez Gene ID:             246595
////
Query:                      lcl|NW_020311285.1_prot_XP_020707989.1_40
Gene:                       dme:Dmel_CG6148 Past1
Entrez Gene ID:             41569
Pathway:                    Endocytosis     KEGG PATHWAY    dme04144
                            Hemostasis      Reactome        R-DME-109582
                            Factors involved in megakaryocyte development and platelet production   Reactome        R-DME-98323
</pre></div>
</div>
<p>If your analysis doesn’t complete as you expected please look at your ‘condor_stderr’ and ‘condor_stdout’ files. If that doesn’t clarify the problem contact us at <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">agbase<span>&#64;</span>email<span>&#46;</span>arizona<span>&#46;</span>edu</a> or <a class="reference external" href="mailto:support&#37;&#52;&#48;cyverse&#46;org">support<span>&#64;</span>cyverse<span>&#46;</span>org</a>.</p>
</section>
</section>
<section id="id2">
<h2><strong>KOBAS identify 3.0.3</strong><a class="headerlink" href="#id2" title="Permalink to this headline"></a></h2>
<section id="id3">
<h3><strong>Launching the App</strong><a class="headerlink" href="#id3" title="Permalink to this headline"></a></h3>
<p><a class="reference internal" href="../_images/kobas_ident.png"><img alt="kobas_ident" src="../_images/kobas_ident.png" style="width: 700px;" /></a></p>
<p><strong>Analysis Name: KOBAS_identify_3.0.3_analysis_1:</strong>
This menu is used to name the job you will run so that you can find it later.
Analysis Name: The default name is “KOBAS_identify_3.0.3_analysis1”. We recommend changing the ‘analysis1’ portion of this to reflect the data you are running.</p>
<p><strong>Comments:</strong>
(Optional) You can add additional information in the comments section to distinguish your analyses further.</p>
<p><strong>Select output folder:</strong>
This is where your results will be placed. The default (recommended) is your ‘analyses’ folder.</p>
<p><strong>Retain Inputs:</strong>
Enabling this flag will copy all the input files into the analysis result folder.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Selecting this option will rapidly consume your allocated space. It is not recommended. Your inputs will always remain available in the folder in which you stored them.</p>
</div>
<section id="id4">
<h4><strong>Input</strong><a class="headerlink" href="#id4" title="Permalink to this headline"></a></h4>
<p><strong>Foreground File:</strong> Use the ‘browse’ button on the right side of the field to navigate to your input file. This should be the output of KOBAS annotate.</p>
<p><strong>Background:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
</section>
<section id="id5">
<h4><strong>Options</strong><a class="headerlink" href="#id5" title="Permalink to this headline"></a></h4>
<p><strong>Cutoff:</strong> Annotation terms with less than cutoff number of genes are not used for statistical tests. Default is 5.</p>
<p><strong>Method:</strong> Choose the statistical method to be used from the drop-down list. Default is hypergeometric/Fisher’s Exact.</p>
<p><strong>FDR:</strong> Method for determining false discovery rate. Default is Benjamnini-Hochberg.</p>
</section>
<section id="id6">
<h4><strong>Output</strong><a class="headerlink" href="#id6" title="Permalink to this headline"></a></h4>
<p><strong>Output File Name:</strong> Provide an output file name .</p>
</section>
</section>
<section id="understanding-your-identify-results">
<h3><strong>Understanding Your Identify Results</strong><a class="headerlink" href="#understanding-your-identify-results" title="Permalink to this headline"></a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><p><strong>logs folder:</strong> This folder contains the ‘conder_stderr’ and ‘condor_stdout’ files. The files record feedback, progress and, importantly, any errors the app encountered during the analysis. You won’t normally need to look at these but they are very helpful in figuring out what may have happened if your output doesn’t look like you expected.</p></li>
<li><p><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</p></li>
<li><p><strong>&lt;output_file_name_you_provided&gt;:</strong> KOBAS identify generates a text file with the name you provide.</p></li>
</ul>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>##Databases: PANTHER, KEGG PATHWAY, Reactome, BioCyc
##Statistical test method: hypergeometric test / Fisher&#39;s exact test
##FDR correction method: Benjamini and Hochberg

#Term       Database        ID      Input number    Background number       P-Value Corrected P-Value       Input   Hyperlink
Hedgehog signaling pathway - fly    KEGG PATHWAY    dme04341        12      33      3.20002656734e-18       1.76001461204e-16       lcl|NW_020311286.1_prot_XP_012256678.1_51|lcl|NW_020311286.1_prot_XP_025602973.1_48|lcl|NW_020311286.1_prot_XP_012256683.1_52|lcl|NW_020311286.1_prot_XP_012256679.1_55|lcl|NW_020311286.1_prot_XP_012256674.1_54|lcl|NW_020311286.1_prot_XP_020708336.1_46|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311286.1_prot_XP_012256682.1_53|lcl|NW_020311286.1_prot_XP_025603025.1_47|lcl|NW_020311286.1_prot_XP_020708334.1_49|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_020708333.1_50 http://www.genome.jp/kegg-bin/show_pathway?dme04341/dme:Dmel_CG6963%09red/dme:Dmel_CG6054%09red
Hedgehog signaling pathway  PANTHER P00025  6       13      3.6166668094e-10        9.94583372585e-09       lcl|NW_020311286.1_prot_XP_025602279.1_78|lcl|NW_020311286.1_prot_XP_025602289.1_76|lcl|NW_020311286.1_prot_XP_025602264.1_79|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_012256943.1_77     http://www.pantherdb.org/pathway/pathwayDiagram.jsp?catAccession=P00025
Signaling by NOTCH2 Reactome        R-DME-1980145   3       8       2.00259649553e-05       0.000275357018136       lcl|NW_020311285.1_prot_XP_012256118.1_28|lcl|NW_020311285.1_prot_XP_012256117.1_27|lcl|NW_020311285.1_prot_XP_012256119.1_26   http://www.reactome.org/cgi-bin/eventbrowser_st_id?ST_ID=R-DME-1980145
</pre></div>
</div>
<p>If your analysis doesn’t complete as you expected please look at your ‘condor_stderr’ and ‘condor_stdout’ files. If that doesn’t clarify the problem contact us at <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">agbase<span>&#64;</span>email<span>&#46;</span>arizona<span>&#46;</span>edu</a> or <a class="reference external" href="mailto:support&#37;&#52;&#48;cyverse&#46;org">support<span>&#64;</span>cyverse<span>&#46;</span>org</a>.</p>
</section>
</section>
<section id="id7">
<h2><strong>KOBAS annotate and identify 3.0.3</strong><a class="headerlink" href="#id7" title="Permalink to this headline"></a></h2>
<section id="id8">
<h3><strong>Launching the App</strong><a class="headerlink" href="#id8" title="Permalink to this headline"></a></h3>
<p><a class="reference internal" href="../_images/kobas_annoident.png"><img alt="kobas_annoident" src="../_images/kobas_annoident.png" style="width: 700px;" /></a></p>
<p>This app runs both the annotate and identify analyses together as a convenience for user who wish to run both steps.</p>
<p><strong>Analysis Name: KOBAS_annotate_and_identify_3.0.3_analysis_1:</strong>
This menu is used to name the job you will run so that you can find it later.
Analysis Name: The default name is “KOBAS_annotate_identify_3.0.3_analysis1”. We recommend changing the ‘analysis1’ portion of this to reflect the data you are running.</p>
<p><strong>Comments:</strong>
(Optional) You can add additional information in the comments section to distinguish your analyses further.</p>
<p><strong>Select output folder:</strong>
This is where your results will be placed. The default (recommended) is your ‘analyses’ folder.</p>
<p><strong>Retain Inputs:</strong>
Enabling this flag will copy all the input files into the analysis result folder.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Selecting this option will rapidly consume your allocated space. It is not recommended. Your inputs will always remain available in the folder in which you stored them.</p>
</div>
<section id="id9">
<h4><strong>Input</strong><a class="headerlink" href="#id9" title="Permalink to this headline"></a></h4>
<p><strong>Input File:</strong> Use the ‘browse’ button on the right side of the field to navigate to your input file.</p>
<p><strong>Input File Type:</strong> Select your input file type from the drop-down list. If your file type isn’t there then the app does not support that file type.</p>
</section>
<section id="annotate-options">
<h4><strong>Annotate Options</strong><a class="headerlink" href="#annotate-options" title="Permalink to this headline"></a></h4>
<p><strong>Species Code:</strong> Enter the species for the species of the sequences in your input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you don’t know the code for your species it can be found here: <a class="reference external" href="https://www.kegg.jp/kegg/catalog/org_list.html">https://www.kegg.jp/kegg/catalog/org_list.html</a></p>
<p>If your species of interest is not available then you should choose the code for the closest-related species available</p>
</div>
<p><strong>E value:</strong> This is the evalue to use in the BLAST search. The default is 1e-5.</p>
<p><strong>Rank:</strong> rank cutoff for valid hits from BLAST result. Default is 5.</p>
<p><strong>Covergage:</strong> subject coverage cutoff for BLAST. Default is 0.</p>
<p><strong>Ortholog:</strong> when checked KOBAS will only use orthologs for cross species annotation.</p>
</section>
<section id="identify-options">
<h4><strong>Identify Options</strong><a class="headerlink" href="#identify-options" title="Permalink to this headline"></a></h4>
<p><strong>Cutoff:</strong> Annotation terms with less than cutoff number of genes are not used for statistical tests. Default is 5.</p>
<p><strong>Method:</strong> Choose the statistical method to be used from the drop-down list. Default is hypergeometric/Fisher’s Exact.</p>
<p><strong>FDR:</strong> Method for determining false discovery rate. Default is Benjamnini-Hochberg.</p>
</section>
<section id="id10">
<h4><strong>Output</strong><a class="headerlink" href="#id10" title="Permalink to this headline"></a></h4>
<p><strong>Output File Basename:</strong> This will the the prefix of your output files.</p>
</section>
</section>
<section id="understanding-your-annotate-and-identify-pipeline-results">
<h3><strong>Understanding Your Annotate and Identify Pipeline Results</strong><a class="headerlink" href="#understanding-your-annotate-and-identify-pipeline-results" title="Permalink to this headline"></a></h3>
<p>If all goes well, you should get the following:</p>
<ul class="simple">
<li><p><strong>logs folder:</strong> This folder contains the ‘conder_stderr’ and ‘condor_stdout’ files. The files record feedback, progress and, importantly, any errors the app encountered during the analysis. You won’t normally need to look at these but they are very helpful in figuring out what may have happened if your output doesn’t look like you expected.</p></li>
<li><p><strong>sqlite3 folder:</strong> This folder contains the annotation database files used in your analysis</p></li>
<li><p><strong>seq_pep folder:</strong> This folder contains the BLAST database files used in your analysis.</p></li>
<li><p><strong>&lt;species&gt;.tsv:</strong> This is the tab-delimited output from the BLAST search. It is unlikely that you will need to look at this file.</p></li>
<li><p><strong>&lt;basename&gt;_annotate_out.txt:</strong> KOBAS annotate generates a text file with the name you provide. It has two sections.</p></li>
</ul>
<p>The first sections looks like this:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>##dme       Drosophila melanogaster (fruit fly)
##Method: BLAST     Options: evalue &lt;= 1e-05
##Summary:  87 succeed, 0 fail

#Query      Gene ID|Gene name|Hyperlink
lcl|NW_020311285.1_prot_XP_012256083.1_15   dme:Dmel_CG34349|Unc-13-4B|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG34349
lcl|NW_020311286.1_prot_XP_020708336.1_46   dme:Dmel_CG6963|gish|http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<p>The second section follows a dashed line and looks like this:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>--------------------

////
Query:                      lcl|NW_020311285.1_prot_XP_012256083.1_15
Gene:                       dme:Dmel_CG34349        Unc-13-4B
Entrez Gene ID:             43002
////
Query:                      lcl|NW_020311286.1_prot_XP_020708336.1_46
Gene:                       dme:Dmel_CG6963 gish
Entrez Gene ID:             49701
Pathway:                    Hedgehog signaling pathway - fly        KEGG PATHWAY    dme04341
////
Query:                      lcl|NW_020311285.1_prot_XP_020707987.1_39
Gene:                       dme:Dmel_CG30403
Entrez Gene ID:             246595
////
Query:                      lcl|NW_020311285.1_prot_XP_020707989.1_40
Gene:                       dme:Dmel_CG6148 Past1
Entrez Gene ID:             41569
Pathway:                    Endocytosis     KEGG PATHWAY    dme04144
                            Hemostasis      Reactome        R-DME-109582
                            Factors involved in megakaryocyte development and platelet production   Reactome        R-DME-98323mel_CG6963
lcl|NW_020311285.1_prot_XP_020707987.1_39   dme:Dmel_CG30403||http://www.genome.jp/dbget-bin/www_bget?dme:Dmel_CG30403
</pre></div>
</div>
<ul class="simple">
<li><p><strong>&lt;basename&gt;_identify_out.txt:</strong> KOBAS identify generates a text file with the name you provide.</p></li>
</ul>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>##Databases: PANTHER, KEGG PATHWAY, Reactome, BioCyc
##Statistical test method: hypergeometric test / Fisher&#39;s exact test
##FDR correction method: Benjamini and Hochberg

#Term       Database        ID      Input number    Background number       P-Value Corrected P-Value       Input   Hyperlink
Hedgehog signaling pathway - fly    KEGG PATHWAY    dme04341        12      33      3.20002656734e-18       1.76001461204e-16       lcl|NW_020311286.1_prot_XP_012256678.1_51|lcl|NW_020311286.1_prot_XP_025602973.1_48|lcl|NW_020311286.1_prot_XP_012256683.1_52|lcl|NW_020311286.1_prot_XP_012256679.1_55|lcl|NW_020311286.1_prot_XP_012256674.1_54|lcl|NW_020311286.1_prot_XP_020708336.1_46|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311286.1_prot_XP_012256682.1_53|lcl|NW_020311286.1_prot_XP_025603025.1_47|lcl|NW_020311286.1_prot_XP_020708334.1_49|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_020708333.1_50 http://www.genome.jp/kegg-bin/show_pathway?dme04341/dme:Dmel_CG6963%09red/dme:Dmel_CG6054%09red
Hedgehog signaling pathway  PANTHER P00025  6       13      3.6166668094e-10        9.94583372585e-09       lcl|NW_020311286.1_prot_XP_025602279.1_78|lcl|NW_020311286.1_prot_XP_025602289.1_76|lcl|NW_020311286.1_prot_XP_025602264.1_79|lcl|NW_020311285.1_prot_XP_012256108.1_32|lcl|NW_020311285.1_prot_XP_012256109.1_33|lcl|NW_020311286.1_prot_XP_012256943.1_77     http://www.pantherdb.org/pathway/pathwayDiagram.jsp?catAccession=P00025
Signaling by NOTCH2 Reactome        R-DME-1980145   3       8       2.00259649553e-05       0.000275357018136       lcl|NW_020311285.1_prot_XP_012256118.1_28|lcl|NW_020311285.1_prot_XP_012256117.1_27|lcl|NW_020311285.1_prot_XP_012256119.1_26   http://www.reactome.org/cgi-bin/eventbrowser_st_id?ST_ID=R-DME-1980145
</pre></div>
</div>
<p>If your analysis doesn’t complete as you expected please look at your ‘condor_stderr’ and ‘condor_stdout’ files. If that doesn’t clarify the problem contact us at <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">agbase<span>&#64;</span>email<span>&#46;</span>arizona<span>&#46;</span>edu</a> or <a class="reference external" href="mailto:support&#37;&#52;&#48;cyverse&#46;org">support<span>&#64;</span>cyverse<span>&#46;</span>org</a>.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="intro.html" class="btn btn-neutral float-left" title="Intro" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="using_kobas_cmd.html" class="btn btn-neutral float-right" title="KOBAS on the Command Line" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Fiona M. McCarthy.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>