

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>KOBAS on the Command Line &mdash; AgBase 1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="AgBase 1.0 documentation" href="../index.html"/>
        <link rel="prev" title="KOBAS on CyVerse" href="using_kobas_in_cyverse_DE.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> AgBase
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Agbase Home</span></p>
<ul class="simple">
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../workflow/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">TransDecoder</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">GOanna</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../goanna/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../goanna/using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">InterProScan</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
</ul>
<p class="caption"><span class="caption-text">Combine GAFs</span></p>
<ul class="simple">
</ul>
<p class="caption"><span class="caption-text">KOBAS</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>KOBAS on the Command Line</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-databases"><strong>Getting the databases</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#container-technologies"><strong>Container Technologies</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-kobas-using-docker"><strong>Running KOBAS using Docker</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-kobas-container"><strong>Getting the KOBAS container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-help-and-usage-statement"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-goanna-with-data"><strong>Running GOanna with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-command"><strong>Example Command</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#breakdown-of-command"><strong>Breakdown of Command</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#understanding-your-results"><strong>Understanding Your Results</strong></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#running-goanna-using-singularity"><strong>Running GOanna using Singularity</strong></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-the-goanna-container"><strong>Getting the GOanna container</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2"><strong>Getting the Help and Usage Statement</strong></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3"><strong>Running GOanna with Data</strong></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-pbs-script"><strong>Example PBS Script</strong></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id4"><strong>Breakdown of Command</strong></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#id5"><strong>Understanding Your Results</strong></a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">AgBase</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li><strong>KOBAS on the Command Line</strong></li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/kobas/using_kobas_cmd.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="kobas-on-the-command-line">
<h1><strong>KOBAS on the Command Line</strong><a class="headerlink" href="#kobas-on-the-command-line" title="Permalink to this headline">¶</a></h1>
<div class="section" id="getting-the-databases">
<h2><strong>Getting the databases</strong><a class="headerlink" href="#getting-the-databases" title="Permalink to this headline">¶</a></h2>
<p>To run the tool you need some public data. The files can be downloaded directly fro the <a class="reference external" href="kobas.cbi.pku.edu.cn">KOBAS homepage</a>. These directories are also available as two tar archives in the CyVerse Data Store. The files are best downloaded with <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html">iCommands</a>. Once iCommands is <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html#icommands-first-time-configuration">setup</a> you can use ‘iget’ to download the data.</p>
<ol class="arabic simple">
<li>seq_pep.tar: species-specific BLAST databases used by KOBAS</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>iget /iplant/home/shared/iplantcollaborative/protein_blast_dbs/kobas/seq_pep.tar

tar -xf seq_pep.tar
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li>sqlite3.tar: species-specific annotation databases used by KOBAS</li>
</ol>
<div class="highlight-bash"><div class="highlight"><pre><span></span>iget /iplant/home/shared/iplantcollaborative/protein_blast_dbs/kobas/sqlite3.tar

tar -xf sqlite3.tar
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The above commands should result in two directories (seq_pep and sqlite3) each containing many files. There is no need to unzip the .gz files.</p>
</div>
</div>
<div class="section" id="container-technologies">
<h2><strong>Container Technologies</strong><a class="headerlink" href="#container-technologies" title="Permalink to this headline">¶</a></h2>
<p>KOBAS is provided as a Docker container.</p>
<p>A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another.</p>
<p>There are two major containerization technologies: <strong>Docker</strong> and <strong>Singularity</strong>.</p>
<p>Docker containers can be run with either technology.</p>
</div>
<div class="section" id="running-kobas-using-docker">
<h2><strong>Running KOBAS using Docker</strong><a class="headerlink" href="#running-kobas-using-docker" title="Permalink to this headline">¶</a></h2>
<div class="admonition-about-docker admonition">
<p class="first admonition-title">About Docker</p>
<ul class="last simple">
<li>Docker must be installed on the computer you wish to use for your analysis.</li>
<li>To run Docker you must have ‘root’ permissions (or use sudo).</li>
<li>Docker will run all containers as ‘root’. This makes Docker incompatible with HPC systems (see Singularity below).</li>
<li>Docker can be run on your local computer, a server, a cloud virtual machine (such as CyVerse Atmosphere) etc. Docker can be installed quickly on an Atmosphere instance by typing ‘ezd’.</li>
<li>For more information on installing Docker on other systems see this tutorial:  <a class="reference external" href="https://learning.cyverse.org/projects/container_camp_workshop_2019/en/latest/docker/dockerintro.html">Installing Docker on your machine</a>.</li>
</ul>
</div>
<div class="section" id="getting-the-kobas-container">
<h3><strong>Getting the KOBAS container</strong><a class="headerlink" href="#getting-the-kobas-container" title="Permalink to this headline">¶</a></h3>
<p>The GOanna tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/kobas">GOanna container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>docker pull agbase/kobas:3.0.3_0
</pre></div>
</div>
<div class="admonition-remember admonition">
<p class="first admonition-title">Remember</p>
<p>You must have root permissions or use sudo, like so:</p>
<p class="last">sudo docker pull agbase/kobas:3.0.3_0</p>
</div>
</div>
<div class="section" id="getting-the-help-and-usage-statement">
<h3><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#getting-the-help-and-usage-statement" title="Permalink to this headline">¶</a></h3>
<div class="highlight-bash"><div class="highlight"><pre><span></span>sudo docker run --rm -v <span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>:/work-dir agbase/kobas:3.0.3_0 -h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>Options:
[-h prints this help statement]

[-a runs KOBAS annotate]
KOBAS annotate options:
    -i INFILE can be FASTA or one-per-lineidentifiers. See -t intype for details.
    -s SPECIES 3 or 4 letter species abbreviation (can be found here: ftp://ftp.cbi.pku.edu.cn/pub/KOBAS_3.0_DOWNLOAD/species_abbr.txt or here: https://www.kegg.jp/kegg/catalog/org_list.html)
    -o OUTPUT file (Default is stdout.)
    [-l LIST available species, or list available databases for a specific species]
    [-t INTYPE (fasta:pro, fasta:nuc, blastout:xml, blastout:tab, id:ncbigi, id:uniprot, id:ensembl, id:ncbigene), default fasta:pro]
    [-e EVALUE expect threshold for BLAST, default 1e-5]
    [-r RANK rank cutoff for valid hits from BLAST result, default is 5]
    [-C COVERAGE subject coverage cutoff for BLAST, default 0]
    [-z ORTHOLOG whether only use orthologs for cross-species annotation or not, default NO (if only using orthologs, please provide the species abbreviation of your input)]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/, This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]
    [-T number of THREADS to use in BLAST search. Default = 8]

[-g runs KOBAS identify]
    KOBAS identify options:
    -f FGFILE foreground file, the output of annotate
    -b BGFILE background file, species abbreviation, see this list for species codes: https://www.kegg.jp/kegg/catalog/org_list.html
    -o OUTPUT file (Default is stdout.)
    [-d DB databases for selection, 1-letter abbreviation separated by &quot;/&quot;: K for KEGG PATHWAY, n for PID, b for BioCarta, R for Reactome, B for BioCyc, p for PANTHER,
           o for OMIM, k for KEGG DISEASE, f for FunDO, g for GAD, N for NHGRI GWAS Catalog and G for Gene Ontology, default K/n/b/R/B/p/o/k/f/g/N/]
    [-m METHOD choose statistical test method: b for binomial test, c for chi-square test, h for hypergeometric test / Fisher&#39;s exact test, and x for frequency list,
           default hypergeometric test / Fisher&#39;s exact test
    [-n FDR choose false discovery rate (FDR) correction method: BH for Benjamini and Hochberg, BY for Benjamini and Yekutieli, QVALUE, and None, default BH
    [-c CUTOFF terms with less than cutoff number of genes are not used for statistical tests, default 5]
    [-k KOBAS HOME The path to kobas_home, which is the parent directory of sqlite3/ and seq_pep/. This is the absolute path in the container.]
    [-v BLAST HOME The path to blast_home, which is the parent directory of blastx and blastp. This is the absolute path in the container.]
    [-y BLASTDB The path to seq_pep/. This is the absolute path in the container.]
    [-q KOBASDB The path to sqlite3/. This is the absolute path in the container.]
    [-p BLASTP The path to blastp. This is the absolute path in the container.]
    [-x BLASTX The path to blastx. This is the absolute path in the container.]

[-j runs both KOBAS annotate and identify]
</pre></div>
</div>
</div>
<div class="section" id="running-goanna-with-data">
<h3><strong>Running GOanna with Data</strong><a class="headerlink" href="#running-goanna-with-data" title="Permalink to this headline">¶</a></h3>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There are 3 directories built into this container. These directories should be used to mount data.</p>
<ul class="last simple">
<li>/agbase_database</li>
<li>/go_info</li>
<li>/work-dir</li>
</ul>
</div>
<p>GOanna has three required parameters:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>-a BLAST database basename <span class="o">(</span>acceptable options are listed in the help/usage<span class="o">)</span>
-c peptide FASTA file to BLAST
-o output file basename
</pre></div>
</div>
<div class="section" id="example-command">
<h4><strong>Example Command</strong><a class="headerlink" href="#example-command" title="Permalink to this headline">¶</a></h4>
<div class="highlight-none"><div class="highlight"><pre><span></span>sudo docker run \
--rm \
-v /home/amcooksey/i5k/agbase_database:/agbase_database \
-v /home/amcooksey/i5k/go_info:/go_info \
-v $(pwd):/work-dir \
agbase/goanna:2.0 \
-a invertebrates \
-c AROS_10.faa \
-o AROS_10_invert_exponly \
-p \
-g 70 \
-s 900 \
-d RefSeq \
-u &quot;Amanda Cooksey&quot; \
-x 37344
</pre></div>
</div>
</div>
<div class="section" id="breakdown-of-command">
<h4><strong>Breakdown of Command</strong><a class="headerlink" href="#breakdown-of-command" title="Permalink to this headline">¶</a></h4>
<p><strong>sudo docker run:</strong> tells docker to run</p>
<p><strong>&#8211;rm:</strong> removes the container when the analysis has finished. The image will remain for future use.</p>
<p><strong>-v /home/amcooksey/i5k/agbase_database:/agbase_database:</strong> tells docker to mount the &#8216;agbase_database&#8217; directory I downloaded to the host machine to the &#8216;/agbase_database&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-v /home/amcooksey/i5k/go_info:/go_info:</strong> mounts &#8216;go_info&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-v $(pwd):/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>agbase/goanna:2.0:</strong> the name of the Docker image to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are GOanna options</p>
</div>
<p><strong>-a invertebrates:</strong> GOanna BLAST database to use&#8211;first of three required options.</p>
<p><strong>-c AROS_10.faa:</strong> input file (peptide FASTA)&#8211;second of three required options</p>
<p><strong>-o AROS_10_invert_exponly:</strong> output file basename&#8211;last of three required options</p>
<p><strong>-p:</strong> our input file has NCBI deflines. This specifies how to parse them.</p>
<p><strong>-g 70:</strong> tells GOanna to keep only those matches with at least 70% identity</p>
<p><strong>-s 900:</strong> tells GOanna to keep only those matches with a bitscore above 900</p>
<p><strong>-d RefSeq:</strong> database of query ID. This will appear in column 1 of the GAF output file.</p>
<p><strong>-u &#8220;Amanda Cooksey&#8221;:</strong> name to appear in column 15 of the GAF output file</p>
<p><strong>-x 37344:</strong> NCBI taxon ID of input file species will appear in column 13 of the GAF output file</p>
</div>
</div>
<div class="section" id="understanding-your-results">
<h3><strong>Understanding Your Results</strong><a class="headerlink" href="#understanding-your-results" title="Permalink to this headline">¶</a></h3>
<p>If all goes well, you should get 4 output files:</p>
<p><strong>&lt;basename&gt;.asn:</strong> This is standard BLAST output format that allows for conversion to other formats. You probably won’t need to look at this output.</p>
<p><strong>&lt;basename&gt;.html:</strong> This output displays in your web browser so that you can view pairwise alignments to determine BLAST parameters.</p>
<p><strong>&lt;basename&gt;.tsv:</strong> This is the tab-delimited BLAST output that can be opened and sorted in Excel to determine BLAST parameter values. The file contains the following columns:</p>
<ul class="simple">
<li>Query ID</li>
<li>query length</li>
<li>query start</li>
<li>query end</li>
<li>subject ID</li>
<li>subject length</li>
<li>subject start</li>
<li>subject end</li>
<li>e-value</li>
<li>percent ID</li>
<li>query coverage</li>
<li>percent positive ID</li>
<li>gap openings</li>
<li>total gaps</li>
<li>bitscore</li>
<li>raw score</li>
</ul>
<p>For more information on the BLAST output parameters see the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279684/#_appendices_Options_for_the_commandline_a_.">NCBI BLAST documentation</a>.</p>
<p><strong>&lt;basename&gt;_goanna_gaf.tsv:</strong> This is the standard tab-separated <a class="reference external" href="http://geneontology.org/docs/go-annotation-file-gaf-format-2.1">GO annotation file format</a>  that is used by the GO Consortium and by software tools that accept GO annotation files to do GO enrichment.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</div>
</div>
<div class="section" id="running-goanna-using-singularity">
<h2><strong>Running GOanna using Singularity</strong><a class="headerlink" href="#running-goanna-using-singularity" title="Permalink to this headline">¶</a></h2>
<div class="admonition-about-singularity admonition">
<p class="first admonition-title">About Singularity</p>
<ul class="last simple">
<li>does not require ‘root’ permissions</li>
<li>runs all containers as the user that is logged into the host machine</li>
<li>HPC systems are likely to have Singularity installed and are unlikely to object if asked to install it (no guarantees).</li>
<li>can be run on any machine where is is installed</li>
<li>more information about <a class="reference external" href="https://singularity.lbl.gov/docs-installation">installing Singularity</a></li>
<li>This tool was tested using Singularity 3.0. Users with Singularity 2.x will need to modify the commands accordingly.</li>
</ul>
</div>
<div class="admonition-hpc-job-schedulers admonition">
<p class="first admonition-title">HPC Job Schedulers</p>
<p class="last">Although Singularity can be installed on any computer this documentation assumes it will be run on an HPC system. The tool was tested on a PBSPro system and the job submission scripts below reflect that. Submission scripts will need to be modified for use with other job scheduler systems.</p>
</div>
<div class="section" id="getting-the-goanna-container">
<h3><strong>Getting the GOanna container</strong><a class="headerlink" href="#getting-the-goanna-container" title="Permalink to this headline">¶</a></h3>
<p>The GOanna tool is available as a Docker container on Docker Hub:
<a class="reference external" href="https://hub.docker.com/r/agbase/goanna">GOanna container</a></p>
<p>The container can be pulled with this command:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>singularity pull docker://agbase/goanna:2.0
</pre></div>
</div>
</div>
<div class="section" id="id2">
<h3><strong>Getting the Help and Usage Statement</strong><a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p><strong>Example PBS script:</strong></p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N goanna</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/GOanna

singularity pull docker://agbase/goanna:2.0

singularity run <span class="se">\</span>
goanna_2.0.sif <span class="se">\</span>
-h
</pre></div>
</div>
<div class="highlight-none"><div class="highlight"><pre><span></span>Options:
-a BLAST database basename (&#39;arthropod&#39;, &#39;bacteria&#39;, &#39;bird&#39;, &#39;crustacean&#39;, &#39;fish&#39;, &#39;fungi&#39;, &#39;human&#39;, &#39;insecta&#39;,
   &#39;invertebrates&#39;, &#39;mammals&#39;, &#39;nematode&#39;, &#39;plants&#39;, &#39;rodents&#39; &#39;uniprot_sprot&#39;, &#39;uniprot_trembl&#39;, &#39;vertebrates&#39;
    or &#39;viruses&#39;)
-c peptide fasta filename
-o BLAST output file basename
[-b transfer GO with experimental evidence only (&#39;yes&#39; or &#39;no&#39;). Default = &#39;yes&#39;.]
[-d database of query ID. If your entry contains spaces either substitute and underscore (_) or,
    to preserve the space, use quotes around your entry. Default: &#39;user_input_db&#39;]
[-e Expect value (E) for saving hits. Default is 10.]
[-f Number of aligned sequences to keep. Default: 3]
[-g BLAST percent identity above which match should be kept. Default: keep all matches.]
[-h help]
[-m BLAST percent positive identity above which match should be kept. Default: keep all matches.]
[-s bitscore above which match should be kept. Default: keep all matches.]
[-k Maximum number of gap openings allowed for match to be kept.Default: 100]
[-l Maximum number of total gaps allowed for match to be kept. Default: 1000]
[-q Minimum query coverage per subject for match to be kept. Default: keep all matches]
[-t Number of threads.  Default: 8]
[-u &#39;Assigned by&#39; field of your GAF output file. If your entry contains spaces (eg. firstname lastname)
    either substitute and underscore (_) or, to preserve the space, use quotes around your entry (eg. &quot;firstname lastname&quot;)
    Default: &#39;user&#39;]
[-x Taxon ID of the query species. Default: &#39;taxon:0000&#39;]
[-p parse_deflines. Parse query and subject bar delimited sequence identifiers]
</pre></div>
</div>
</div>
<div class="section" id="id3">
<h3><strong>Running GOanna with Data</strong><a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p>There are 3 directories built into this container. These directories should be used to mount data.</p>
<ul class="last simple">
<li>/agbase_database</li>
<li>/go_info</li>
<li>/work-dir</li>
</ul>
</div>
<p>GOanna has three required parameters:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>-a BLAST database basename <span class="o">(</span>acceptable options are listed in the help/usage<span class="o">)</span>
-c peptide FASTA file to BLAST
-o output file basename
</pre></div>
</div>
<div class="section" id="example-pbs-script">
<h4><strong>Example PBS Script</strong><a class="headerlink" href="#example-pbs-script" title="Permalink to this headline">¶</a></h4>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#PBS -N goanna</span>
<span class="c1">#PBS -W group_list=fionamcc</span>
<span class="c1">#PBS -l select=1:ncpus=28:mem=168gb</span>
<span class="c1">#PBS -q standard</span>
<span class="c1">#PBS -l walltime=6:0:0</span>
<span class="c1">#PBS -l cput=168:0:0</span>

module load singularity

<span class="nb">cd</span> /rsgrps/shaneburgess/amanda/i5k/GOanna

singularity pull docker://agbase/goanna:2.0

singularity run <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/agbase_database:/agbase_database <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/go_info:/go_info <span class="se">\</span>
-B /rsgrps/shaneburgess/amanda/i5k/goanna:/work-dir <span class="se">\</span>
goanna_2.0.sif <span class="se">\</span>
-a invertebrates <span class="se">\</span>
-c AROS_10.faa <span class="se">\</span>
-o AROS_10_invert_exponly <span class="se">\</span>
-p <span class="se">\</span>
-g <span class="m">70</span> <span class="se">\</span>
-s <span class="m">900</span> <span class="se">\</span>
-d RefSeq <span class="se">\</span>
-u <span class="s2">&quot;Amanda Cooksey&quot;</span> <span class="se">\</span>
-x <span class="m">37344</span> <span class="se">\</span>
-t 28
</pre></div>
</div>
</div>
<div class="section" id="id4">
<h4><strong>Breakdown of Command</strong><a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h4>
<p><strong>singularity run:</strong> tells Singularity to run</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/agbase_database:/agbase_database:</strong> tells docker to mount the &#8216;agbase_database&#8217; directory I downloaded to the host machine to the &#8216;/agbase_database&#8217; directory within the container. The syntax for this is: &lt;absolute path on host&gt;:&lt;absolute path in container&gt;</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/go_info:/go_info:</strong> mounts &#8216;go_info&#8217; directory on host machine into &#8216;go_info&#8217; directory inside the container</p>
<p><strong>-B /rsgrps/shaneburgess/amanda/i5k/goanna:/work-dir:</strong> mounts my current working directory on the host machine to &#8216;/work-dir&#8217; in the container</p>
<p><strong>goanna_2.0.sif:</strong> the name of the Singularity image file to use</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">All the options supplied after the image name are GOanna options</p>
</div>
<p><strong>-a invertebrates:</strong> GOanna BLAST database to use&#8211;first of three required options.</p>
<p><strong>-c AROS_10.faa:</strong> input file (peptide FASTA)&#8211;second of three required options</p>
<p><strong>-o AROS_10_invert_exponly:</strong> output file basename&#8211;last of three required options</p>
<p><strong>-p:</strong> our input file has NCBI deflines. This specifies how to parse them.</p>
<p><strong>-g 70:</strong> tells GOanna to keep only those matches with at least 70% identity</p>
<p><strong>-s 900:</strong> tells GOanna to keep only those matches with a bitscore above 900</p>
<p><strong>-d RefSeq:</strong> database of query ID. This will appear in column 1 of the GAF output file.</p>
<p><strong>-u &#8220;Amanda Cooksey&#8221;:</strong> name to appear in column 15 of the GAF output file</p>
<p><strong>-x 37344:</strong> NCBI taxon ID of input file species will appear in column 13 of the GAF output file</p>
<p><strong>-t 28:</strong> number of threads to use for BLAST. This was run on a node with 28 cores.</p>
</div>
</div>
<div class="section" id="id5">
<h3><strong>Understanding Your Results</strong><a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<p>If all goes well, you should get 4 output files:</p>
<p><strong>&lt;basename&gt;.asn:</strong> This is standard BLAST output format that allows for conversion to other formats. You probably won’t need to look at this output.</p>
<p><strong>&lt;basename&gt;.html:</strong> This output displays in your web browser so that you can view pairwise alignments to determine BLAST parameters.</p>
<p><strong>&lt;basename&gt;.tsv:</strong> This is the tab-delimited BLAST output that can be opened and sorted in Excel to determine BLAST parameter values. The file contains the following columns:</p>
<ul class="simple">
<li>Query ID</li>
<li>query length</li>
<li>query start</li>
<li>query end</li>
<li>subject ID</li>
<li>subject length</li>
<li>subject start</li>
<li>subject end</li>
<li>e-value</li>
<li>percent ID</li>
<li>query coverage</li>
<li>percent positive ID</li>
<li>gap openings</li>
<li>total gaps</li>
<li>bitscore</li>
<li>raw score</li>
</ul>
<p>For more information on the BLAST output parameters see the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279684/#_appendices_Options_for_the_commandline_a_.">NCBI BLAST documentation</a>.</p>
<p><strong>&lt;basename&gt;_goanna_gaf.tsv:</strong> This is the standard tab-separated <a class="reference external" href="http://geneontology.org/docs/go-annotation-file-gaf-format-2.1">GO annotation file format</a>  that is used by the GO Consortium and by software tools that accept GO annotation files to do GO enrichment.</p>
<p>If you see more files in your output folder there may have been an error in the analysis or there may have been no GO to transfer. <a class="reference external" href="mailto:agbase&#37;&#52;&#48;email&#46;arizona&#46;edu">Contact us</a>.</p>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="using_kobas_in_cyverse_DE.html" class="btn btn-neutral" title="KOBAS on CyVerse" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Fiona M. McCarthy.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>