<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Intro &mdash; AgBase 1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="GOanna on CyVerse" href="using_goanna_in_cyverse_DE.html" />
    <link rel="prev" title="Functional Annotation Workflow" href="../agbase/workflow.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            AgBase
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Agbase Home</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../agbase/workflow.html"><strong>Functional Annotation Workflow</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">GOanna</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Intro</strong></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#where-to-find-goanna"><strong>Where to Find GOanna</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#getting-the-goanna-databases"><strong>Getting the GOanna Databases</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="#help-and-usage-statement"><strong>Help and Usage Statement</strong></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="using_goanna_in_cyverse_DE.html"><strong>GOanna on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_goanna_cmd.html"><strong>GOanna on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="using_goanna_ceres.html"><strong>GOanna on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">InterProScan</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_in_cyverse_DE.html"><strong>InterProScan on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_cmd.html"><strong>InterProScan on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../interproscan/using_iprs_ceres.html"><strong>InterProScan on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Combine GAFs</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cyverse.html"><strong>Combine GAFs on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_cmd.html"><strong>Combine GAFs on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../combine_gafs/combine_gafs_ceres.html"><strong>Combine GAFs on the ARS Ceres HPC</strong></a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">KOBAS</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../kobas/intro.html"><strong>Intro</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_in_cyverse_DE.html"><strong>KOBAS on CyVerse</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_cmd.html"><strong>KOBAS on the Command Line</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="../kobas/using_kobas_ceres.html"><strong>KOBAS on the ARS Ceres HPC</strong></a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">AgBase</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active"><strong>Intro</strong></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/goanna/intro.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="intro">
<h1><strong>Intro</strong><a class="headerlink" href="#intro" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p>GOanna performs a BLAST search, allows you to filter based on BLAST match parameters and transfers Gene Ontology (GO) functional annotations from the BLAST matches to your input genes.</p></li>
<li><p>GOanna accepts a protein FASTA file as input.</p></li>
<li><p>BLAST databases are created by AgBase based upon proteins that have GO available and subsetted by phyla. We recommend selecting the database most closely related to the sequence used as input.</p></li>
<li><p>We strongly recommend selecting only GO annotations based on experimental evidence codes. This will ensure the best quality annotations for your data.</p></li>
<li><p>The remaining parameters are standard BLAST parameters. More information on determining the best BLAST parameters for your specific data set can be found in the section below.</p></li>
</ul>
<p>Results and analysis from the application of GOanna to the Official gene set v3.0 protein set from <em>Diaphorina citri</em> followed by a differential expression analysis was presented at a seminar in the University of Arizona Animal and Comparative Biomedical Sciences in Fall 2020. The <a class="reference external" href="https://www.slideshare.net/suryasaha/functional-annotation-of-invertebrate-genomes">slides</a> and <a class="reference external" href="https://arizona.zoom.us/rec/play/tZZ-fuutrj43T9fBtASDAaR9W9S0fP6s1XQbrvQOz0e0VnYHYVL1MOMaZ-F4v45qOmXQkV1MUXQ7tufD">video</a> are available online.</p>
<section id="where-to-find-goanna">
<h2><strong>Where to Find GOanna</strong><a class="headerlink" href="#where-to-find-goanna" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://hub.docker.com/r/agbase/goanna">Docker Hub</a></p></li>
<li><p><a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=354731ae-71ab-11e9-b82a-008cfa5ae621&amp;system-id=de">CyVerse Discovery Environment</a></p></li>
<li><p><a class="reference external" href="https://agbase.arizona.edu/cgi-bin/tools/GOanna.cgi">AgBase</a></p></li>
</ul>
</section>
<section id="getting-the-goanna-databases">
<h2><strong>Getting the GOanna Databases</strong><a class="headerlink" href="#getting-the-goanna-databases" title="Permalink to this headline"></a></h2>
<p>To run the tool you need some public data. These files are now available as gzipped files to aid downloading. The directories are best downloaded with <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html">iCommands</a>. Once iCommands is <a class="reference external" href="https://cyverse-data-store-guide.readthedocs-hosted.com/en/latest/step2.html#icommands-first-time-configuration">setup</a> you can use ‘iget’ to download the data.</p>
<ol class="arabic simple">
<li><p>agbase_database: species subset to run BLAST against  (this command will download the entire directory)</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>iget -rPT /iplant/home/shared/iplantcollaborative/protein_blast_dbs/agbase_database
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>go_info: Uniprot GO annotations (this command will download the entire directory)</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>iget -rPT /iplant/home/shared/iplantcollaborative/protein_blast_dbs/go_info
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Each of these tools accepts a peptide FASTA file. For those users with nucloetide sequences some documentation has been provided for using <strong>TransDecoder</strong> (although other tools are also acceptable).
The <a class="reference external" href="https://de.cyverse.org/de/?type=apps&amp;app-id=74828a18-f351-11e8-be2b-008cfa5ae621&amp;system-id=de">TransDecoder app</a> is available through CyVerse or as a <a class="reference external" href="https://quay.io/repository/biocontainers/transdecoder?tab=tags">BioContainer</a> for use on the command line.</p>
</div>
</section>
<section id="help-and-usage-statement">
<span id="goannausage"></span><h2><strong>Help and Usage Statement</strong><a class="headerlink" href="#help-and-usage-statement" title="Permalink to this headline"></a></h2>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Options:
-a BLAST database basename (&#39;arthropod&#39;, &#39;bacteria&#39;, &#39;bird&#39;, &#39;crustacean&#39;, &#39;fish&#39;, &#39;fungi&#39;, &#39;human&#39;, &#39;insecta&#39;,
   &#39;invertebrates&#39;, &#39;mammals&#39;, &#39;nematode&#39;, &#39;plants&#39;, &#39;rodents&#39; &#39;uniprot_sprot&#39;, &#39;uniprot_trembl&#39;, &#39;vertebrates&#39;
    or &#39;viruses&#39;)
-c peptide fasta filename
-o output file basename
[-b transfer GO with experimental evidence only (&#39;yes&#39; or &#39;no&#39;). Default = &#39;yes&#39;.]
[-d database of query ID. If your entry contains spaces either substitute and underscore (_) or,
    to preserve the space, use quotes around your entry. Default: &#39;user_input_db&#39;]
[-e Expect value (E) for saving hits. Default is 10.]
[-f Number of aligned sequences to keep. Default: 3]
[-g BLAST percent identity above which match should be kept. Default: keep all matches.]
[-h help]
[-m BLAST percent positive identity above which match should be kept. Default: keep all matches.]
[-s bitscore above which match should be kept. Default: keep all matches.]
[-k Maximum number of gap openings allowed for match to be kept.Default: 100]
[-l Maximum number of total gaps allowed for match to be kept. Default: 1000]
[-q Minimum query coverage per subject for match to be kept. Default: keep all matches]
[-t Number of threads.  Default: 8]
[-u &#39;Assigned by&#39; field of your GAF output file. If your entry contains spaces (eg. firstname lastname)
    either substitute and underscore (_) or, to preserve the space, use quotes around your entry (eg. &quot;firstname lastname&quot;)
    Default: &#39;user&#39;]
[-x Taxon ID of the query species. Default: &#39;taxon:0000&#39;]
[-p parse_deflines. Parse query and subject bar delimited sequence identifiers]
</pre></div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../agbase/workflow.html" class="btn btn-neutral float-left" title="Functional Annotation Workflow" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="using_goanna_in_cyverse_DE.html" class="btn btn-neutral float-right" title="GOanna on CyVerse" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2019, Fiona M. McCarthy.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>